// @alfresco/js-api/esm5/src/api/content-custom-api/api/base.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
var BaseApi = (function () {
    function BaseApi(alfrescoApi) {
        this.alfrescoApi = alfrescoApi;
    }
    Object.defineProperty(BaseApi.prototype, "apiClientPrivate", {
        get: function () {
            return this.alfrescoApi.contentPrivateClient;
        },
        enumerable: false,
        configurable: true
    });
    Object.defineProperty(BaseApi.prototype, "apiClient", {
        get: function () {
            return this.alfrescoApi.contentClient;
        },
        enumerable: false,
        configurable: true
    });
    BaseApi.prototype.post = function (options) {
        return this.apiClientPrivate.post(options);
    };
    BaseApi.prototype.put = function (options) {
        return this.apiClientPrivate.put(options);
    };
    BaseApi.prototype.get = function (options) {
        return this.apiClientPrivate.get(options);
    };
    BaseApi.prototype.delete = function (options) {
        return this.apiClientPrivate.delete(options);
    };
    BaseApi.prototype.errorMessage = function (param, methodName) {
        return "Missing param ".concat(param, " in ").concat(methodName);
    };
    return BaseApi;
}());
export { BaseApi };
