// @alfresco/js-api/src/api/content-rest-api/api/index.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIS = void 0;
const tslib_1 = require("tslib");
(0, tslib_1.__exportStar)(require("./actions.api"), exports);
const actions_api_1 = require("./actions.api");
(0, tslib_1.__exportStar)(require("./activities.api"), exports);
const activities_api_1 = require("./activities.api");
(0, tslib_1.__exportStar)(require("./audit.api"), exports);
const audit_api_1 = require("./audit.api");
(0, tslib_1.__exportStar)(require("./comments.api"), exports);
const comments_api_1 = require("./comments.api");
(0, tslib_1.__exportStar)(require("./downloads.api"), exports);
const downloads_api_1 = require("./downloads.api");
(0, tslib_1.__exportStar)(require("./favorites.api"), exports);
const favorites_api_1 = require("./favorites.api");
(0, tslib_1.__exportStar)(require("./groups.api"), exports);
const groups_api_1 = require("./groups.api");
(0, tslib_1.__exportStar)(require("./networks.api"), exports);
const networks_api_1 = require("./networks.api");
(0, tslib_1.__exportStar)(require("./nodes.api"), exports);
const nodes_api_1 = require("./nodes.api");
(0, tslib_1.__exportStar)(require("./people.api"), exports);
const people_api_1 = require("./people.api");
(0, tslib_1.__exportStar)(require("./preferences.api"), exports);
const preferences_api_1 = require("./preferences.api");
(0, tslib_1.__exportStar)(require("./probes.api"), exports);
const probes_api_1 = require("./probes.api");
(0, tslib_1.__exportStar)(require("./queries.api"), exports);
const queries_api_1 = require("./queries.api");
(0, tslib_1.__exportStar)(require("./ratings.api"), exports);
const ratings_api_1 = require("./ratings.api");
(0, tslib_1.__exportStar)(require("./renditions.api"), exports);
const renditions_api_1 = require("./renditions.api");
(0, tslib_1.__exportStar)(require("./sharedlinks.api"), exports);
const sharedlinks_api_1 = require("./sharedlinks.api");
(0, tslib_1.__exportStar)(require("./sites.api"), exports);
const sites_api_1 = require("./sites.api");
(0, tslib_1.__exportStar)(require("./tags.api"), exports);
const tags_api_1 = require("./tags.api");
(0, tslib_1.__exportStar)(require("./trashcan.api"), exports);
const trashcan_api_1 = require("./trashcan.api");
(0, tslib_1.__exportStar)(require("./versions.api"), exports);
const versions_api_1 = require("./versions.api");
exports.APIS = {
    ActionsApi: actions_api_1.ActionsApi,
    ActivitiesApi: activities_api_1.ActivitiesApi,
    AuditApi: audit_api_1.AuditApi,
    CommentsApi: comments_api_1.CommentsApi,
    DownloadsApi: downloads_api_1.DownloadsApi,
    FavoritesApi: favorites_api_1.FavoritesApi,
    GroupsApi: groups_api_1.GroupsApi,
    NetworksApi: networks_api_1.NetworksApi,
    NodesApi: nodes_api_1.NodesApi,
    PeopleApi: people_api_1.PeopleApi,
    PreferencesApi: preferences_api_1.PreferencesApi,
    ProbesApi: probes_api_1.ProbesApi,
    QueriesApi: queries_api_1.QueriesApi,
    RatingsApi: ratings_api_1.RatingsApi,
    RenditionsApi: renditions_api_1.RenditionsApi,
    SharedlinksApi: sharedlinks_api_1.SharedlinksApi,
    SitesApi: sites_api_1.SitesApi,
    TagsApi: tags_api_1.TagsApi,
    TrashcanApi: trashcan_api_1.TrashcanApi,
    VersionsApi: versions_api_1.VersionsApi
};
