// @alfresco/js-api/src/api/content-rest-api/api/nodes.api.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodesApi = void 0;
const associationEntry_1 = require("../model/associationEntry");
const childAssociationEntry_1 = require("../model/childAssociationEntry");
const directAccessUrlEntry_1 = require("../model/directAccessUrlEntry");
const nodeAssociationPaging_1 = require("../model/nodeAssociationPaging");
const nodeChildAssociationPaging_1 = require("../model/nodeChildAssociationPaging");
const nodeEntry_1 = require("../model/nodeEntry");
const base_api_1 = require("./base.api");
const assert_1 = require("../../../assert");
const alfrescoApiClient_1 = require("../../../alfrescoApiClient");
class NodesApi extends base_api_1.BaseApi {
    copyNode(nodeId, nodeBodyCopy, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(nodeBodyCopy, 'nodeBodyCopy');
        opts = opts || {};
        const postBody = nodeBodyCopy;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/copy', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeEntry_1.NodeEntry);
    }
    createAssociation(nodeId, associationBodyCreate, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(associationBodyCreate, 'associationBodyCreate');
        opts = opts || {};
        const postBody = associationBodyCreate;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/targets', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, associationEntry_1.AssociationEntry);
    }
    createNode(nodeId, nodeBodyCreate, opts, formParams) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(nodeBodyCreate, 'nodeBodyCreate');
        opts = opts || {};
        let postBody = nodeBodyCreate;
        let pathParams = {
            'nodeId': nodeId
        };
        let queryParams = {
            'autoRename': opts['autoRename'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        let headerParams = {};
        formParams = formParams || {};
        let contentTypes, returnType = null;
        if (formParams.filedata) {
            contentTypes = ['multipart/form-data'];
        }
        else {
            contentTypes = ['application/json'];
            returnType = nodeEntry_1.NodeEntry;
        }
        let accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/children', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, returnType);
    }
    createSecondaryChildAssociation(nodeId, secondaryChildAssociationBodyCreate, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(secondaryChildAssociationBodyCreate, 'secondaryChildAssociationBodyCreate');
        opts = opts || {};
        const postBody = secondaryChildAssociationBodyCreate;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/secondary-children', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, childAssociationEntry_1.ChildAssociationEntry);
    }
    deleteAssociation(nodeId, targetId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(targetId, 'targetId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'targetId': targetId
        };
        const queryParams = {
            'assocType': opts['assocType']
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/targets/{targetId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteNode(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'permanent': opts['permanent']
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteSecondaryChildAssociation(nodeId, childId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(childId, 'childId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'childId': childId
        };
        const queryParams = {
            'assocType': opts['assocType']
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/secondary-children/{childId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getNode(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeEntry_1.NodeEntry);
    }
    getNodeContent(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'attachment': opts['attachment']
        };
        const headerParams = {
            'If-Modified-Since': opts['ifModifiedSince'], 'Range': opts['range']
        };
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/octet-stream'];
        return this.apiClient.callApi('/nodes/{nodeId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, 'blob');
    }
    listNodeChildren(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'orderBy': (0, alfrescoApiClient_1.buildCollectionParam)(opts['orderBy'], 'csv'),
            'where': opts['where'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'includeSource': opts['includeSource'],
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/children', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeChildAssociationPaging_1.NodeChildAssociationPaging);
    }
    listParents(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'where': opts['where'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'includeSource': opts['includeSource'],
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/parents', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeAssociationPaging_1.NodeAssociationPaging);
    }
    listSecondaryChildren(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'where': opts['where'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'includeSource': opts['includeSource'],
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/secondary-children', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeChildAssociationPaging_1.NodeChildAssociationPaging);
    }
    listSourceAssociations(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'where': opts['where'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/sources', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeAssociationPaging_1.NodeAssociationPaging);
    }
    listTargetAssociations(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'where': opts['where'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/targets', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeAssociationPaging_1.NodeAssociationPaging);
    }
    lockNode(nodeId, nodeBodyLock, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(nodeBodyLock, 'nodeBodyLock');
        opts = opts || {};
        const postBody = nodeBodyLock;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/lock', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeEntry_1.NodeEntry);
    }
    moveNode(nodeId, nodeBodyMove, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(nodeBodyMove, 'nodeBodyMove');
        opts = opts || {};
        const postBody = nodeBodyMove;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/move', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeEntry_1.NodeEntry);
    }
    unlockNode(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/unlock', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeEntry_1.NodeEntry);
    }
    updateNode(nodeId, nodeBodyUpdate, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(nodeBodyUpdate, 'nodeBodyUpdate');
        opts = opts || {};
        const postBody = nodeBodyUpdate;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeEntry_1.NodeEntry);
    }
    updateNodeContent(nodeId, contentBodyUpdate, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(contentBodyUpdate, 'contentBodyUpdate');
        opts = opts || {};
        const postBody = contentBodyUpdate;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'majorVersion': opts['majorVersion'],
            'comment': opts['comment'],
            'name': opts['name'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/octet-stream'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/content', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, nodeEntry_1.NodeEntry);
    }
    requestDirectAccessUrl(nodeId) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        const pathParams = {
            'nodeId': nodeId
        };
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/request-direct-access-url', 'POST', pathParams, {}, {}, {}, null, contentTypes, accepts, directAccessUrlEntry_1.DirectAccessUrlEntry);
    }
}
exports.NodesApi = NodesApi;
