// @alfresco/js-api/src/api/content-rest-api/api/tags.api.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagsApi = void 0;
const tagEntry_1 = require("../model/tagEntry");
const tagPaging_1 = require("../model/tagPaging");
const base_api_1 = require("./base.api");
const assert_1 = require("../../../assert");
const alfrescoApiClient_1 = require("../../../alfrescoApiClient");
class TagsApi extends base_api_1.BaseApi {
    createTagForNode(nodeId, tagBodyCreate, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(tagBodyCreate, 'tagBodyCreate');
        opts = opts || {};
        const postBody = tagBodyCreate;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/tags', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, tagEntry_1.TagEntry);
    }
    deleteTagFromNode(nodeId, tagId) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(tagId, 'tagId');
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'tagId': tagId
        };
        const queryParams = {};
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/tags/{tagId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getTag(tagId, opts) {
        (0, assert_1.throwIfNotDefined)(tagId, 'tagId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'tagId': tagId
        };
        const queryParams = {
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/tags/{tagId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, tagEntry_1.TagEntry);
    }
    listTags(opts) {
        opts = opts || {};
        const postBody = null;
        const pathParams = {};
        const queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv'),
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/tags', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, tagPaging_1.TagPaging);
    }
    listTagsForNode(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/tags', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, tagPaging_1.TagPaging);
    }
    updateTag(tagId, tagBodyUpdate, opts) {
        (0, assert_1.throwIfNotDefined)(tagId, 'tagId');
        (0, assert_1.throwIfNotDefined)(tagBodyUpdate, 'tagBodyUpdate');
        opts = opts || {};
        const postBody = tagBodyUpdate;
        const pathParams = {
            'tagId': tagId
        };
        const queryParams = {
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/tags/{tagId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, tagEntry_1.TagEntry);
    }
}
exports.TagsApi = TagsApi;
