// @alfresco/js-api/src/api/content-rest-api/api/versions.api.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.VersionsApi = void 0;
const directAccessUrlEntry_1 = require("../model/directAccessUrlEntry");
const renditionEntry_1 = require("../model/renditionEntry");
const renditionPaging_1 = require("../model/renditionPaging");
const versionEntry_1 = require("../model/versionEntry");
const versionPaging_1 = require("../model/versionPaging");
const base_api_1 = require("./base.api");
const assert_1 = require("../../../assert");
const alfrescoApiClient_1 = require("../../../alfrescoApiClient");
class VersionsApi extends base_api_1.BaseApi {
    createVersionRendition(nodeId, versionId, renditionBodyCreate) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        (0, assert_1.throwIfNotDefined)(renditionBodyCreate, 'renditionBodyCreate');
        const postBody = renditionBodyCreate;
        const pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        const queryParams = {};
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/renditions', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteVersion(nodeId, versionId) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        const queryParams = {};
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getVersion(nodeId, versionId) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        const queryParams = {};
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, versionEntry_1.VersionEntry);
    }
    getVersionContent(nodeId, versionId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        const queryParams = {
            'attachment': opts['attachment']
        };
        const headerParams = {
            'If-Modified-Since': opts['ifModifiedSince'], 'Range': opts['range']
        };
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/octet-stream'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, 'blob');
    }
    getVersionRendition(nodeId, versionId, renditionId) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        (0, assert_1.throwIfNotDefined)(renditionId, 'renditionId');
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'versionId': versionId, 'renditionId': renditionId
        };
        const queryParams = {};
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/renditions/{renditionId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, renditionEntry_1.RenditionEntry);
    }
    getVersionRenditionContent(nodeId, versionId, renditionId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        (0, assert_1.throwIfNotDefined)(renditionId, 'renditionId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'versionId': versionId, 'renditionId': renditionId
        };
        const queryParams = {
            'attachment': opts['attachment'],
            'placeholder': opts['placeholder']
        };
        const headerParams = {
            'If-Modified-Since': opts['ifModifiedSince'], 'Range': opts['range']
        };
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/octet-stream'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/renditions/{renditionId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, 'blob');
    }
    listVersionHistory(nodeId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId
        };
        const queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv'),
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems']
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, versionPaging_1.VersionPaging);
    }
    listVersionRenditions(nodeId, versionId, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        opts = opts || {};
        const postBody = null;
        const pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        const queryParams = {
            'where': opts['where']
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/renditions', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, renditionPaging_1.RenditionPaging);
    }
    revertVersion(nodeId, versionId, revertBody, opts) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        (0, assert_1.throwIfNotDefined)(revertBody, 'revertBody');
        opts = opts || {};
        const postBody = revertBody;
        const pathParams = {
            'nodeId': nodeId, 'versionId': versionId
        };
        const queryParams = {
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        const headerParams = {};
        const formParams = {};
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/revert', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, versionEntry_1.VersionEntry);
    }
    requestDirectAccessUrl(nodeId, versionId) {
        (0, assert_1.throwIfNotDefined)(nodeId, 'nodeId');
        (0, assert_1.throwIfNotDefined)(versionId, 'versionId');
        const pathParams = {
            'nodeId': nodeId,
            'versionId': versionId
        };
        const contentTypes = ['application/json'];
        const accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/versions/{versionId}/request-direct-access-url', 'POST', pathParams, {}, {}, {}, null, contentTypes, accepts, directAccessUrlEntry_1.DirectAccessUrlEntry);
    }
}
exports.VersionsApi = VersionsApi;
