// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/api/AlfrescoApiActiviti.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { IntegrationAlfrescoCloudApi } from '../../../../api/activiti-rest-api/api/integrationAlfrescoCloud.api';
import { IntegrationAlfrescoOnPremiseApi } from '../../../../api/activiti-rest-api/api/integrationAlfrescoOnPremise.api';
var AlfrescoApiActiviti = (function () {
    function AlfrescoApiActiviti() {
    }
    AlfrescoApiActiviti.prototype.init = function (alfrescoApi) {
        this.integrationAlfrescoCloudApi = new IntegrationAlfrescoCloudApi(alfrescoApi);
        this.IntegrationAlfrescoOnPremiseApi = new IntegrationAlfrescoOnPremiseApi(alfrescoApi);
    };
    AlfrescoApiActiviti.prototype.confirmAuthorisation = function (code) {
        return this.integrationAlfrescoCloudApi.confirmAuthorisation(code);
    };
    AlfrescoApiActiviti.prototype.getAllNetworks = function () {
        return this.integrationAlfrescoCloudApi.getAllNetworks();
    };
    AlfrescoApiActiviti.prototype.getAllSites = function (networkId) {
        return this.integrationAlfrescoCloudApi.getAllSites(networkId);
    };
    AlfrescoApiActiviti.prototype.getAllSitesByRepositoryId = function (repositoryId) {
        return this.IntegrationAlfrescoOnPremiseApi.getAllSites(repositoryId);
    };
    AlfrescoApiActiviti.prototype.getContentInFolder = function (networkId, folderId) {
        return this.integrationAlfrescoCloudApi.getContentInFolder(networkId, folderId);
    };
    AlfrescoApiActiviti.prototype.getContentInFolderByRepositoryId = function (repositoryId, folderId) {
        return this.IntegrationAlfrescoOnPremiseApi.getContentInFolder(repositoryId, folderId);
    };
    AlfrescoApiActiviti.prototype.getContentInSite = function (networkId, siteId) {
        return this.integrationAlfrescoCloudApi.getContentInSite(networkId, siteId);
    };
    AlfrescoApiActiviti.prototype.getContentInSiteByRepositoryId = function (repositoryId, siteId) {
        return this.IntegrationAlfrescoOnPremiseApi.getContentInSite(repositoryId, siteId);
    };
    AlfrescoApiActiviti.prototype.getRepositories = function (opts) {
        return this.IntegrationAlfrescoOnPremiseApi.getRepositories(opts);
    };
    return AlfrescoApiActiviti;
}());
export { AlfrescoApiActiviti };
