// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/api/AppsApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { AppDefinitionsApi } from '../../../../api/activiti-rest-api/api/appDefinitions.api';
import { RuntimeAppDefinitionsApi } from '../../../../api/activiti-rest-api/api/runtimeAppDefinitions.api';
var AppsApi = (function () {
    function AppsApi() {
    }
    AppsApi.prototype.init = function (alfrescoApi) {
        this.runtimeAppDefinitionsApi = new RuntimeAppDefinitionsApi(alfrescoApi);
        this.appDefinitionsApi = new AppDefinitionsApi(alfrescoApi);
    };
    AppsApi.prototype.deployAppDefinitions = function (saveObject) {
        return this.runtimeAppDefinitionsApi.deployAppDefinitions(saveObject);
    };
    AppsApi.prototype.exportAppDefinition = function (modelId) {
        return this.appDefinitionsApi.exportAppDefinition(modelId);
    };
    AppsApi.prototype.getAppDefinitions = function () {
        return this.runtimeAppDefinitionsApi.getAppDefinitions();
    };
    AppsApi.prototype.importAppDefinition = function (file) {
        return this.appDefinitionsApi.importAppDefinition(file);
    };
    AppsApi.prototype.importNewAppDefinition = function (modelId, file) {
        return this.appDefinitionsApi.updateAppDefinition(modelId, file);
    };
    AppsApi.prototype.publishAppDefinition = function (modelId, publishModel) {
        return this.appDefinitionsApi.publishAppDefinition(modelId, publishModel);
    };
    AppsApi.prototype.updateAppDefinition = function (modelId, updatedModel) {
        return this.appDefinitionsApi.updateAppDefinition(modelId, updatedModel);
    };
    return AppsApi;
}());
export { AppsApi };
