// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/api/AppsDefinitionApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { AppDefinitionsApi } from '../../../../api/activiti-rest-api/api/appDefinitions.api';
var AppsDefinitionApi = (function () {
    function AppsDefinitionApi() {
    }
    AppsDefinitionApi.prototype.init = function (alfrescoApi) {
        this.appDefinitionsApi = new AppDefinitionsApi(alfrescoApi);
    };
    AppsDefinitionApi.prototype.exportAppDefinition = function (modelId) {
        return this.appDefinitionsApi.exportAppDefinition(modelId);
    };
    AppsDefinitionApi.prototype.importAppDefinition = function (file, opts) {
        return this.appDefinitionsApi.importAppDefinition(file, opts);
    };
    AppsDefinitionApi.prototype.importNewAppDefinition = function (modelId, file) {
        return this.appDefinitionsApi.updateAppDefinition(modelId, file);
    };
    AppsDefinitionApi.prototype.publishAppDefinition = function (modelId, publishModel) {
        return this.appDefinitionsApi.publishAppDefinition(modelId, publishModel);
    };
    return AppsDefinitionApi;
}());
export { AppsDefinitionApi };
