// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/api/ContentApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { ContentApi as NewContentApi } from '../../../../api/activiti-rest-api/api/content.api';
import { ProcessInstancesApi } from '../../../../api/activiti-rest-api/api/processInstances.api';
var ContentApi = (function () {
    function ContentApi() {
    }
    ContentApi.prototype.init = function (alfrescoApi) {
        this.contentApi = new NewContentApi(alfrescoApi);
        this.processInstancesApi = new ProcessInstancesApi(alfrescoApi);
    };
    ContentApi.prototype.createRelatedContentOnProcessInstancePostBody = function (processInstanceId, relatedContent, opts) {
        return this.contentApi.createRelatedContentOnProcessInstance(processInstanceId, relatedContent, opts);
    };
    ContentApi.prototype.createRelatedContentOnProcessInstance = function (processInstanceId, file, opts) {
        return this.contentApi.createRelatedContentOnProcessInstance(processInstanceId, file, opts);
    };
    ContentApi.prototype.createRelatedContentOnTaskPostBody = function (taskId, relatedContent, opts) {
        return this.contentApi.createRelatedContentOnTask(taskId, relatedContent, opts);
    };
    ContentApi.prototype.createRelatedContentOnTask = function (taskId, file, opts) {
        return this.contentApi.createRelatedContentOnTask(taskId, file, opts);
    };
    ContentApi.prototype.createTemporaryRawRelatedContent = function (file) {
        return this.contentApi.createTemporaryRawRelatedContent(file);
    };
    ContentApi.prototype.createTemporaryRelatedContent = function (relatedContent) {
        return this.contentApi.createTemporaryRelatedContent(relatedContent);
    };
    ContentApi.prototype.deleteContent = function (contentId) {
        return this.contentApi.deleteContent(contentId);
    };
    ContentApi.prototype.getContent = function (contentId) {
        return this.contentApi.getContent(contentId);
    };
    ContentApi.prototype.getProcessInstanceContent = function (processInstanceId) {
        return this.processInstancesApi.getProcessInstanceContent(processInstanceId);
    };
    ContentApi.prototype.getRawContent = function (contentId) {
        return this.contentApi.getRawContent(contentId);
    };
    ContentApi.prototype.getContentPreview = function (contentId) {
        return this.contentApi.getRawContent(contentId, 'preview');
    };
    ContentApi.prototype.getRawContentUrl = function (contentId) {
        return this.contentApi.getRawContentUrl(contentId);
    };
    ContentApi.prototype.getContentThumbnail = function (contentId) {
        return this.contentApi.getRawContent(contentId, 'thumbnail');
    };
    ContentApi.prototype.getRelatedContentForProcessInstance = function (processInstanceId, isRelatedContent) {
        return this.contentApi.getRelatedContentForProcessInstance(processInstanceId, { isRelatedContent: isRelatedContent });
    };
    ContentApi.prototype.getRelatedContentForTask = function (taskId, isRelatedContent) {
        return this.contentApi.getRelatedContentForTask(taskId, { isRelatedContent: isRelatedContent });
    };
    return ContentApi;
}());
export { ContentApi };
