// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/api/EditorApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { FormModelsApi } from '../../../../api/activiti-rest-api/api/formModels.api';
var EditorApi = (function () {
    function EditorApi() {
    }
    EditorApi.prototype.init = function (alfrescoApi) {
        this.formModelsApi = new FormModelsApi(alfrescoApi);
    };
    EditorApi.prototype.getFormHistory = function (formId, formHistoryId) {
        return this.formModelsApi.getFormHistory(formId, formHistoryId);
    };
    EditorApi.prototype.getForm = function (formId) {
        return this.formModelsApi.getForm(formId);
    };
    EditorApi.prototype.getForms = function (input) {
        return this.formModelsApi.getForms(input);
    };
    EditorApi.prototype.saveForm = function (formId, saveRepresentation) {
        return this.formModelsApi.saveForm(formId, saveRepresentation);
    };
    EditorApi.prototype.validateModel = function (formId, saveRepresentation) {
        return this.formModelsApi.validateModel(formId, saveRepresentation);
    };
    return EditorApi;
}());
export { EditorApi };
