// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/api/ProcessDefinitionsFormApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { ProcessDefinitionsApi } from '../../../../api/activiti-rest-api/api/processDefinitions.api';
var ProcessDefinitionsFormApi = (function () {
    function ProcessDefinitionsFormApi() {
    }
    ProcessDefinitionsFormApi.prototype.init = function (alfrescoApi) {
        this.processDefinitionsApi = new ProcessDefinitionsApi(alfrescoApi);
    };
    ProcessDefinitionsFormApi.prototype.getProcessDefinitionStartForm = function (processDefinitionId) {
        return this.processDefinitionsApi.getProcessDefinitionStartForm(processDefinitionId);
    };
    ProcessDefinitionsFormApi.prototype.getRestFieldValues = function (processDefinitionId, field) {
        return this.processDefinitionsApi.getRestFieldValues(processDefinitionId, field);
    };
    ProcessDefinitionsFormApi.prototype.getRestTableFieldValues = function (processDefinitionId, field, column) {
        return this.processDefinitionsApi.getRestTableFieldValues(processDefinitionId, field, column);
    };
    return ProcessDefinitionsFormApi;
}());
export { ProcessDefinitionsFormApi };
