// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/api/ProcessApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { ProcessInstancesApi } from '../../../../api/activiti-rest-api/api/processInstances.api';
import { ProcessDefinitionsApi } from '../../../../api/activiti-rest-api/api/processDefinitions.api';
var ProcessApi = (function () {
    function ProcessApi() {
    }
    ProcessApi.prototype.init = function (alfrescoApi) {
        this.processInstancesApi = new ProcessInstancesApi(alfrescoApi);
        this.processDefinitionsApi = new ProcessDefinitionsApi(alfrescoApi);
    };
    ProcessApi.prototype.deleteProcessInstance = function (processInstanceId) {
        return this.processInstancesApi.deleteProcessInstance(processInstanceId);
    };
    ProcessApi.prototype.filterProcessInstances = function (filterRequest) {
        return this.processInstancesApi.filterProcessInstances(filterRequest);
    };
    ProcessApi.prototype.getProcessDefinitionStartForm = function (processDefinitionId) {
        return this.processDefinitionsApi.getProcessDefinitionStartForm(processDefinitionId);
    };
    ProcessApi.prototype.getProcessDefinitions = function (opts) {
        return this.processDefinitionsApi.getProcessDefinitions(opts);
    };
    ProcessApi.prototype.getProcessInstanceContent = function (processInstanceId) {
        return this.processInstancesApi.getProcessInstanceContent(processInstanceId);
    };
    ProcessApi.prototype.getProcessInstanceStartForm = function (processInstanceId) {
        return this.processInstancesApi.getProcessInstanceStartForm(processInstanceId);
    };
    ProcessApi.prototype.getProcessInstance = function (processInstanceId) {
        return this.processInstancesApi.getProcessInstance(processInstanceId);
    };
    ProcessApi.prototype.getProcessInstances = function (processInstancesQuery) {
        return this.processInstancesApi.getProcessInstances(processInstancesQuery);
    };
    ProcessApi.prototype.getProcessAuditJson = function (processInstanceId) {
        return this.processInstancesApi.getTaskAuditLog(processInstanceId);
    };
    ProcessApi.prototype.getProcessAuditPdf = function (processInstanceId) {
        return this.processInstancesApi.getProcessAuditPdf(processInstanceId);
    };
    ProcessApi.prototype.getRestFieldValues = function (processDefinitionId, field) {
        return this.processDefinitionsApi.getRestFieldValues(processDefinitionId, field);
    };
    ProcessApi.prototype.getRestTableFieldValues = function (processDefinitionId, field, column) {
        return this.processDefinitionsApi.getRestTableFieldValues(processDefinitionId, field, column);
    };
    ProcessApi.prototype.startNewProcessInstance = function (startRequest) {
        return this.processInstancesApi.startNewProcessInstance(startRequest);
    };
    return ProcessApi;
}());
export { ProcessApi };
