// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/api/ProcessInstancesApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { ProcessInstancesApi as NewProcessInstancesApi } from '../../../../api/activiti-rest-api/api';
import { ActivitiCommentsApi } from '../../../../api/activiti-rest-api/api/comments.api';
var ProcessInstancesApi = (function () {
    function ProcessInstancesApi() {
    }
    ProcessInstancesApi.prototype.init = function (alfrescoApi) {
        this.processInstancesApi = new NewProcessInstancesApi(alfrescoApi);
        this.activitiCommentsApi = new ActivitiCommentsApi(alfrescoApi);
    };
    ProcessInstancesApi.prototype.addProcessInstanceComment = function (commentRequest, processInstanceId) {
        return this.activitiCommentsApi.addProcessInstanceComment(commentRequest, processInstanceId);
    };
    ProcessInstancesApi.prototype.deleteProcessInstance = function (processInstanceId) {
        return this.processInstancesApi.deleteProcessInstance(processInstanceId);
    };
    ProcessInstancesApi.prototype.getProcessInstanceComments = function (processInstanceId, opts) {
        return this.activitiCommentsApi.getProcessInstanceComments(processInstanceId, opts);
    };
    ProcessInstancesApi.prototype.getProcessInstanceStartForm = function (processInstanceId) {
        return this.processInstancesApi.getProcessInstanceStartForm(processInstanceId);
    };
    ProcessInstancesApi.prototype.getProcessInstance = function (processInstanceId) {
        return this.processInstancesApi.getProcessInstance(processInstanceId);
    };
    return ProcessInstancesApi;
}());
export { ProcessInstancesApi };
