// @alfresco/js-api/src/api/content-rest-api/model/nodeBodyCreate.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeBodyCreate = void 0;
const associationBody_1 = require("./associationBody");
const childAssociationBody_1 = require("./childAssociationBody");
const definition_1 = require("./definition");
const nodeBodyCreateAssociation_1 = require("./nodeBodyCreateAssociation");
const permissionsBody_1 = require("./permissionsBody");
class NodeBodyCreate {
    constructor(input) {
        if (input) {
            Object.assign(this, input);
            this.permissions = input.permissions ? new permissionsBody_1.PermissionsBody(input.permissions) : undefined;
            this.definition = input.definition ? new definition_1.Definition(input.definition) : undefined;
            this.association = input.association ? new nodeBodyCreateAssociation_1.NodeBodyCreateAssociation(input.association) : undefined;
            if (input.secondaryChildren) {
                this.secondaryChildren = input.secondaryChildren.map((item) => {
                    return new childAssociationBody_1.ChildAssociationBody(item);
                });
            }
            if (input.targets) {
                this.targets = input.targets.map((item) => {
                    return new associationBody_1.AssociationBody(item);
                });
            }
        }
    }
}
exports.NodeBodyCreate = NodeBodyCreate;
