// @alfresco/js-api/src/api/content-rest-api/model/nodeAssociation.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeAssociation = void 0;
const associationInfo_1 = require("./associationInfo");
const contentInfo_1 = require("../../content-rest-api/model/contentInfo");
const dateAlfresco_1 = require("../../content-custom-api/model/dateAlfresco");
const definition_1 = require("./definition");
const pathInfo_1 = require("../../content-rest-api/model/pathInfo");
const permissionsInfo_1 = require("./permissionsInfo");
const userInfo_1 = require("../../content-rest-api/model/userInfo");
class NodeAssociation {
    constructor(input) {
        if (input) {
            Object.assign(this, input);
            this.modifiedAt = input.modifiedAt ? dateAlfresco_1.DateAlfresco.parseDate(input.modifiedAt) : undefined;
            this.modifiedByUser = input.modifiedByUser ? new userInfo_1.UserInfo(input.modifiedByUser) : undefined;
            this.createdAt = input.createdAt ? dateAlfresco_1.DateAlfresco.parseDate(input.createdAt) : undefined;
            this.createdByUser = input.createdByUser ? new userInfo_1.UserInfo(input.createdByUser) : undefined;
            this.content = input.content ? new contentInfo_1.ContentInfo(input.content) : undefined;
            this.path = input.path ? new pathInfo_1.PathInfo(input.path) : undefined;
            this.permissions = input.permissions ? new permissionsInfo_1.PermissionsInfo(input.permissions) : undefined;
            this.definition = input.definition ? new definition_1.Definition(input.definition) : undefined;
            this.association = input.association ? new associationInfo_1.AssociationInfo(input.association) : undefined;
        }
    }
}
exports.NodeAssociation = NodeAssociation;
