// @alfresco/js-api/src/api/content-rest-api/model/siteGroup.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteGroup = void 0;
const groupMember_1 = require("./groupMember");
class SiteGroup {
    constructor(input) {
        if (input) {
            Object.assign(this, input);
            this.group = input.group ? new groupMember_1.GroupMember(input.group) : undefined;
        }
    }
}
exports.SiteGroup = SiteGroup;
(function (SiteGroup) {
    SiteGroup.RoleEnum = {
        SiteConsumer: 'SiteConsumer',
        SiteCollaborator: 'SiteCollaborator',
        SiteContributor: 'SiteContributor',
        SiteManager: 'SiteManager'
    };
})(SiteGroup = exports.SiteGroup || (exports.SiteGroup = {}));
