// @alfresco/js-api/src/api-legacy/activiti-rest-api/src/api/AlfrescoApiActiviti.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.AlfrescoApiActiviti = void 0;
const integrationAlfrescoCloud_api_1 = require("../../../../api/activiti-rest-api/api/integrationAlfrescoCloud.api");
const integrationAlfrescoOnPremise_api_1 = require("../../../../api/activiti-rest-api/api/integrationAlfrescoOnPremise.api");
class AlfrescoApiActiviti {
    init(alfrescoApi) {
        this.integrationAlfrescoCloudApi = new integrationAlfrescoCloud_api_1.IntegrationAlfrescoCloudApi(alfrescoApi);
        this.IntegrationAlfrescoOnPremiseApi = new integrationAlfrescoOnPremise_api_1.IntegrationAlfrescoOnPremiseApi(alfrescoApi);
    }
    confirmAuthorisation(code) {
        return this.integrationAlfrescoCloudApi.confirmAuthorisation(code);
    }
    getAllNetworks() {
        return this.integrationAlfrescoCloudApi.getAllNetworks();
    }
    getAllSites(networkId) {
        return this.integrationAlfrescoCloudApi.getAllSites(networkId);
    }
    getAllSitesByRepositoryId(repositoryId) {
        return this.IntegrationAlfrescoOnPremiseApi.getAllSites(repositoryId);
    }
    getContentInFolder(networkId, folderId) {
        return this.integrationAlfrescoCloudApi.getContentInFolder(networkId, folderId);
    }
    getContentInFolderByRepositoryId(repositoryId, folderId) {
        return this.IntegrationAlfrescoOnPremiseApi.getContentInFolder(repositoryId, folderId);
    }
    getContentInSite(networkId, siteId) {
        return this.integrationAlfrescoCloudApi.getContentInSite(networkId, siteId);
    }
    getContentInSiteByRepositoryId(repositoryId, siteId) {
        return this.IntegrationAlfrescoOnPremiseApi.getContentInSite(repositoryId, siteId);
    }
    getRepositories(opts) {
        return this.IntegrationAlfrescoOnPremiseApi.getRepositories(opts);
    }
}
exports.AlfrescoApiActiviti = AlfrescoApiActiviti;
