// @alfresco/js-api/src/api-legacy/activiti-rest-api/src/api/AppsDefinitionApi.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppsDefinitionApi = void 0;
const appDefinitions_api_1 = require("../../../../api/activiti-rest-api/api/appDefinitions.api");
class AppsDefinitionApi {
    init(alfrescoApi) {
        this.appDefinitionsApi = new appDefinitions_api_1.AppDefinitionsApi(alfrescoApi);
    }
    exportAppDefinition(modelId) {
        return this.appDefinitionsApi.exportAppDefinition(modelId);
    }
    importAppDefinition(file, opts) {
        return this.appDefinitionsApi.importAppDefinition(file, opts);
    }
    importNewAppDefinition(modelId, file) {
        return this.appDefinitionsApi.updateAppDefinition(modelId, file);
    }
    publishAppDefinition(modelId, publishModel) {
        return this.appDefinitionsApi.publishAppDefinition(modelId, publishModel);
    }
}
exports.AppsDefinitionApi = AppsDefinitionApi;
