// @alfresco/js-api/src/api-legacy/activiti-rest-api/src/api/ProcessInstancesApi.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProcessInstancesApi = void 0;
const api_1 = require("../../../../api/activiti-rest-api/api");
const comments_api_1 = require("../../../../api/activiti-rest-api/api/comments.api");
class ProcessInstancesApi {
    init(alfrescoApi) {
        this.processInstancesApi = new api_1.ProcessInstancesApi(alfrescoApi);
        this.activitiCommentsApi = new comments_api_1.ActivitiCommentsApi(alfrescoApi);
    }
    addProcessInstanceComment(commentRequest, processInstanceId) {
        return this.activitiCommentsApi.addProcessInstanceComment(commentRequest, processInstanceId);
    }
    deleteProcessInstance(processInstanceId) {
        return this.processInstancesApi.deleteProcessInstance(processInstanceId);
    }
    getProcessInstanceComments(processInstanceId, opts) {
        return this.activitiCommentsApi.getProcessInstanceComments(processInstanceId, opts);
    }
    getProcessInstanceStartForm(processInstanceId) {
        return this.processInstancesApi.getProcessInstanceStartForm(processInstanceId);
    }
    getProcessInstance(processInstanceId) {
        return this.processInstancesApi.getProcessInstance(processInstanceId);
    }
}
exports.ProcessInstancesApi = ProcessInstancesApi;
