// @alfresco/js-api/esm5/src/api/activiti-rest-api/model/formDefinitionRepresentation.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { FormFieldRepresentation } from './formFieldRepresentation';
import { FormJavascriptEventRepresentation } from './formJavascriptEventRepresentation';
import { FormOutcomeRepresentation } from './formOutcomeRepresentation';
import { FormTabRepresentation } from './formTabRepresentation';
import { FormVariableRepresentation } from './formVariableRepresentation';
var FormDefinitionRepresentation = (function () {
    function FormDefinitionRepresentation(input) {
        if (input) {
            Object.assign(this, input);
            if (input.fields) {
                this.fields = input.fields.map(function (item) {
                    return new FormFieldRepresentation(item);
                });
            }
            if (input.javascriptEvents) {
                this.javascriptEvents = input.javascriptEvents.map(function (item) {
                    return new FormJavascriptEventRepresentation(item);
                });
            }
            if (input.outcomes) {
                this.outcomes = input.outcomes.map(function (item) {
                    return new FormOutcomeRepresentation(item);
                });
            }
            if (input.tabs) {
                this.tabs = input.tabs.map(function (item) {
                    return new FormTabRepresentation(item);
                });
            }
            if (input.variables) {
                this.variables = input.variables.map(function (item) {
                    return new FormVariableRepresentation(item);
                });
            }
        }
    }
    return FormDefinitionRepresentation;
}());
export { FormDefinitionRepresentation };
