// @alfresco/js-api/esm5/src/api/activiti-rest-api/model/processInstanceQueryRepresentation.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
var ProcessInstanceQueryRepresentation = (function () {
    function ProcessInstanceQueryRepresentation(input) {
        if (input) {
            Object.assign(this, input);
        }
    }
    return ProcessInstanceQueryRepresentation;
}());
export { ProcessInstanceQueryRepresentation };
(function (ProcessInstanceQueryRepresentation) {
    ProcessInstanceQueryRepresentation.SortEnum = {
        CreatedDesc: 'created-desc',
        CreatedAsc: 'created-asc',
        EndedDesc: 'ended-desc',
        EndedAsc: 'ended-asc'
    };
    ProcessInstanceQueryRepresentation.StateEnum = {
        Running: 'running',
        Completed: 'completed',
        All: 'all'
    };
})(ProcessInstanceQueryRepresentation || (ProcessInstanceQueryRepresentation = {}));
