// @alfresco/js-api/esm5/src/api/activiti-rest-api/model/taskQueryRepresentation.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { DateAlfresco } from '../../content-custom-api/model/dateAlfresco';
var TaskQueryRepresentation = (function () {
    function TaskQueryRepresentation(input) {
        if (input) {
            Object.assign(this, input);
            this.dueAfter = input.dueAfter ? DateAlfresco.parseDate(input.dueAfter) : undefined;
            this.dueBefore = input.dueBefore ? DateAlfresco.parseDate(input.dueBefore) : undefined;
        }
    }
    return TaskQueryRepresentation;
}());
export { TaskQueryRepresentation };
(function (TaskQueryRepresentation) {
    TaskQueryRepresentation.SortEnum = {
        CreatedDesc: 'created-desc',
        CreatedAsc: 'created-asc',
        DueDesc: 'due-desc',
        DueAsc: 'due-asc'
    };
    TaskQueryRepresentation.StateEnum = {
        Active: 'active',
        Completed: 'completed',
        All: 'all'
    };
})(TaskQueryRepresentation || (TaskQueryRepresentation = {}));
