// @alfresco/js-api/esm5/src/api/activiti-rest-api/model/userRepresentation.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { DateAlfresco } from '../../content-custom-api/model/dateAlfresco';
import { GroupRepresentation } from './groupRepresentation';
import { LightAppRepresentation } from './lightAppRepresentation';
var UserRepresentation = (function () {
    function UserRepresentation(input) {
        if (input) {
            Object.assign(this, input);
            if (input.apps) {
                this.apps = input.apps.map(function (item) {
                    return new LightAppRepresentation(item);
                });
            }
            this.created = input.created ? DateAlfresco.parseDate(input.created) : undefined;
            if (input.groups) {
                this.groups = input.groups.map(function (item) {
                    return new GroupRepresentation(item);
                });
            }
            this.lastUpdate = input.lastUpdate ? DateAlfresco.parseDate(input.lastUpdate) : undefined;
            this.latestSyncTimeStamp = input.latestSyncTimeStamp ? DateAlfresco.parseDate(input.latestSyncTimeStamp) : undefined;
            this.primaryGroup = input.primaryGroup ? new GroupRepresentation(input.primaryGroup) : undefined;
        }
    }
    return UserRepresentation;
}());
export { UserRepresentation };
