// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/index.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { ChangesApi } from './api/changesApi';
import { ChildAssociationsApi } from './api/childAssociationsApi';
import { CommentsApi } from './api/commentsApi';
import { FavoritesApi } from './api/favoritesApi';
import { NodesApi } from './api/nodesApi';
import { NetworksApi } from './api/networksApi';
import { RatingsApi } from './api/ratingsApi';
import { PeopleApi } from './api/peopleApi';
import { RenditionsApi } from './api/renditionsApi';
import { QueriesApi } from './api/queriesApi';
import { SharedlinksApi } from './api/sharedlinksApi';
import { SitesApi } from './api/sitesApi';
import { TagsApi } from './api/tagsApi';
import { WebscriptApi } from './api/webscriptApi';
import { DownloadsApi } from './api/downloadsApi';
import { VersionsApi } from './api/versionsApi';
import { ClassesApi } from './api/classesApi';
import { GroupsApi } from './api/groupsApi';
import { CustomModelApi } from './api/customModelApi';
import { NodesApi as _NodesApi } from './api/nodesApi';
import { ChangesApi as _ChangesApi } from './api/changesApi';
import { ChildAssociationsApi as _ChildAssociationsApi } from './api/childAssociationsApi';
import { CommentsApi as _CommentsApi } from './api/commentsApi';
import { FavoritesApi as _FavoritesApi } from './api/favoritesApi';
import { NetworksApi as _NetworksApi } from './api/networksApi';
import { RatingsApi as _RatingsApi } from './api/ratingsApi';
import { PeopleApi as _PeopleApi } from './api/peopleApi';
import { RenditionsApi as _RenditionsApi } from './api/renditionsApi';
import { QueriesApi as _QueriesApi } from './api/queriesApi';
import { SharedlinksApi as _SharedlinksApi } from './api/sharedlinksApi';
import { SitesApi as _SitesApi } from './api/sitesApi';
import { TagsApi as _TagsApi } from './api/tagsApi';
import { WebscriptApi as _WebscriptApi } from './api/webscriptApi';
import { DownloadsApi as _DownloadsApi } from './api/downloadsApi';
import { VersionsApi as _VersionsApi } from './api/versionsApi';
import { ClassesApi as _ClassesApi } from './api/classesApi';
import { GroupsApi as _GroupsApi } from './api/groupsApi';
import { CustomModelApi as _CustomModelApi } from './api/customModelApi';
export var CONTENT_LEGACY_APIS = {
    ChangesApi: ChangesApi,
    ChildAssociationsApi: ChildAssociationsApi,
    FavoritesApi: FavoritesApi,
    NetworksApi: NetworksApi,
    NodesApi: NodesApi,
    CommentsApi: CommentsApi,
    PeopleApi: PeopleApi,
    RatingsApi: RatingsApi,
    RenditionsApi: RenditionsApi,
    QueriesApi: QueriesApi,
    SharedlinksApi: SharedlinksApi,
    SitesApi: SitesApi,
    TagsApi: TagsApi,
    WebscriptApi: WebscriptApi,
    DownloadsApi: DownloadsApi,
    VersionsApi: VersionsApi,
    ClassesApi: ClassesApi,
    GroupsApi: GroupsApi,
    CustomModelApi: CustomModelApi
};
export var Core;
(function (Core) {
    var ChangesApi = (function (_super) {
        __extends(ChangesApi, _super);
        function ChangesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ChangesApi;
    }(_ChangesApi));
    Core.ChangesApi = ChangesApi;
    var ChildAssociationsApi = (function (_super) {
        __extends(ChildAssociationsApi, _super);
        function ChildAssociationsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ChildAssociationsApi;
    }(_ChildAssociationsApi));
    Core.ChildAssociationsApi = ChildAssociationsApi;
    var FavoritesApi = (function (_super) {
        __extends(FavoritesApi, _super);
        function FavoritesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return FavoritesApi;
    }(_FavoritesApi));
    Core.FavoritesApi = FavoritesApi;
    var NetworksApi = (function (_super) {
        __extends(NetworksApi, _super);
        function NetworksApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return NetworksApi;
    }(_NetworksApi));
    Core.NetworksApi = NetworksApi;
    var NodesApi = (function (_super) {
        __extends(NodesApi, _super);
        function NodesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return NodesApi;
    }(_NodesApi));
    Core.NodesApi = NodesApi;
    var CommentsApi = (function (_super) {
        __extends(CommentsApi, _super);
        function CommentsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CommentsApi;
    }(_CommentsApi));
    Core.CommentsApi = CommentsApi;
    var PeopleApi = (function (_super) {
        __extends(PeopleApi, _super);
        function PeopleApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return PeopleApi;
    }(_PeopleApi));
    Core.PeopleApi = PeopleApi;
    var RatingsApi = (function (_super) {
        __extends(RatingsApi, _super);
        function RatingsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return RatingsApi;
    }(_RatingsApi));
    Core.RatingsApi = RatingsApi;
    var RenditionsApi = (function (_super) {
        __extends(RenditionsApi, _super);
        function RenditionsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return RenditionsApi;
    }(_RenditionsApi));
    Core.RenditionsApi = RenditionsApi;
    var QueriesApi = (function (_super) {
        __extends(QueriesApi, _super);
        function QueriesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return QueriesApi;
    }(_QueriesApi));
    Core.QueriesApi = QueriesApi;
    var SharedlinksApi = (function (_super) {
        __extends(SharedlinksApi, _super);
        function SharedlinksApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return SharedlinksApi;
    }(_SharedlinksApi));
    Core.SharedlinksApi = SharedlinksApi;
    var SitesApi = (function (_super) {
        __extends(SitesApi, _super);
        function SitesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return SitesApi;
    }(_SitesApi));
    Core.SitesApi = SitesApi;
    var TagsApi = (function (_super) {
        __extends(TagsApi, _super);
        function TagsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return TagsApi;
    }(_TagsApi));
    Core.TagsApi = TagsApi;
    var WebscriptApi = (function (_super) {
        __extends(WebscriptApi, _super);
        function WebscriptApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return WebscriptApi;
    }(_WebscriptApi));
    Core.WebscriptApi = WebscriptApi;
    var DownloadsApi = (function (_super) {
        __extends(DownloadsApi, _super);
        function DownloadsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return DownloadsApi;
    }(_DownloadsApi));
    Core.DownloadsApi = DownloadsApi;
    var VersionsApi = (function (_super) {
        __extends(VersionsApi, _super);
        function VersionsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return VersionsApi;
    }(_VersionsApi));
    Core.VersionsApi = VersionsApi;
    var ClassesApi = (function (_super) {
        __extends(ClassesApi, _super);
        function ClassesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ClassesApi;
    }(_ClassesApi));
    Core.ClassesApi = ClassesApi;
    var GroupsApi = (function (_super) {
        __extends(GroupsApi, _super);
        function GroupsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return GroupsApi;
    }(_GroupsApi));
    Core.GroupsApi = GroupsApi;
    var CustomModelApi = (function (_super) {
        __extends(CustomModelApi, _super);
        function CustomModelApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CustomModelApi;
    }(_CustomModelApi));
    Core.CustomModelApi = CustomModelApi;
})(Core || (Core = {}));
