// @alfresco/js-api/esm5/src/api/activiti-rest-api/api/content.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { RelatedContentRepresentation } from '../model/relatedContentRepresentation';
import { ResultListDataRepresentationRelatedContentRepresentation } from '../model/resultListDataRepresentationRelatedContentRepresentation';
import { BaseApi } from './base.api';
import { throwIfNotDefined } from '../../../assert';
var ContentApi = (function (_super) {
    __extends(ContentApi, _super);
    function ContentApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ContentApi.prototype.createRelatedContentOnProcessInstance = function (processInstanceId, relatedContent, opts) {
        throwIfNotDefined(processInstanceId, 'processInstanceId');
        throwIfNotDefined(relatedContent, 'relatedContent');
        opts = opts || {};
        var pathParams = {
            'processInstanceId': processInstanceId
        };
        var queryParams = {
            'isRelatedContent': opts['isRelatedContent']
        };
        var headerParams = {};
        var formParams = {};
        var accepts = ['application/json'];
        if (relatedContent instanceof RelatedContentRepresentation) {
            var postBody = relatedContent;
            var contentTypes = ['application/json'];
            return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
        }
        else {
            var postBody = null;
            formParams = {
                'file': relatedContent
            };
            var contentTypes = ['multipart/form-data'];
            return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/raw-content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
        }
    };
    ContentApi.prototype.createRelatedContentOnTask = function (taskId, relatedContent, opts) {
        throwIfNotDefined(taskId, 'taskId');
        throwIfNotDefined(relatedContent, 'relatedContent');
        opts = opts || {};
        var pathParams = {
            'taskId': taskId
        };
        var queryParams = {
            'isRelatedContent': opts['isRelatedContent']
        };
        var headerParams = {};
        var formParams = {};
        var accepts = ['application/json'];
        if (relatedContent instanceof RelatedContentRepresentation) {
            var postBody = relatedContent;
            var contentTypes = ['application/json'];
            return this.apiClient.callApi('/api/enterprise/tasks/{taskId}/content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
        }
        else {
            var postBody = null;
            formParams = {
                'file': relatedContent
            };
            var contentTypes = ['multipart/form-data'];
            return this.apiClient.callApi('/api/enterprise/tasks/{taskId}/raw-content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
        }
    };
    ContentApi.prototype.createTemporaryRawRelatedContent = function (file) {
        throwIfNotDefined(file, 'file');
        var postBody = null;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {
            'file': file
        };
        var contentTypes = ['multipart/form-data'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/content/raw', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
    };
    ContentApi.prototype.createTemporaryRelatedContent = function (relatedContent) {
        throwIfNotDefined(relatedContent, 'relatedContent');
        var postBody = relatedContent;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/content', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
    };
    ContentApi.prototype.deleteContent = function (contentId) {
        throwIfNotDefined(contentId, 'contentId');
        var postBody = null;
        var pathParams = {
            'contentId': contentId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/content/{contentId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ContentApi.prototype.getContent = function (contentId) {
        throwIfNotDefined(contentId, 'contentId');
        var postBody = null;
        var pathParams = {
            'contentId': contentId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/content/{contentId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RelatedContentRepresentation);
    };
    ContentApi.prototype.getRawContentUrl = function (contentId) {
        return "".concat(this.apiClient.basePath, "/api/enterprise/content/").concat(contentId, "/raw");
    };
    ContentApi.prototype.getRawContent = function (contentId, renditionType) {
        throwIfNotDefined(contentId, 'contentId');
        var postBody = null;
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        if (renditionType) {
            var contentTypes = ['application/json'];
            var accepts = ['application/json'];
            var pathParams = {
                'contentId': contentId,
                'renditionType': renditionType
            };
            return this.apiClient.callApi('/api/enterprise/content/{contentId}/rendition/{renditionType}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, 'blob');
        }
        else {
            var contentTypes = ['application/json'];
            var accepts = ['application/json'];
            var pathParams = {
                'contentId': contentId
            };
            return this.apiClient.callApi('/api/enterprise/content/{contentId}/raw', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, 'blob', undefined, 'blob');
        }
    };
    ContentApi.prototype.getRelatedContentForProcessInstance = function (processInstanceId, opts) {
        throwIfNotDefined(processInstanceId, 'processInstanceId');
        opts = opts || {};
        var postBody = null;
        var pathParams = {
            'processInstanceId': processInstanceId
        };
        var queryParams = {
            'isRelatedContent': opts['isRelatedContent']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationRelatedContentRepresentation);
    };
    ContentApi.prototype.getRelatedContentForTask = function (taskId, opts) {
        throwIfNotDefined(taskId, 'taskId');
        opts = opts || {};
        var postBody = null;
        var pathParams = {
            'taskId': taskId
        };
        var queryParams = {
            'isRelatedContent': opts['isRelatedContent']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/tasks/{taskId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationRelatedContentRepresentation);
    };
    return ContentApi;
}(BaseApi));
export { ContentApi };
