// @alfresco/js-api/esm5/src/api/activiti-rest-api/api/processInstanceVariables.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { RestVariable } from '../model/restVariable';
import { BaseApi } from './base.api';
import { throwIfNotDefined } from '../../../assert';
var ProcessInstanceVariablesApi = (function (_super) {
    __extends(ProcessInstanceVariablesApi, _super);
    function ProcessInstanceVariablesApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ProcessInstanceVariablesApi.prototype.createOrUpdateProcessInstanceVariables = function (processInstanceId, restVariables) {
        throwIfNotDefined(processInstanceId, 'processInstanceId');
        throwIfNotDefined(restVariables, 'restVariables');
        var postBody = restVariables;
        var pathParams = {
            'processInstanceId': processInstanceId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/variables', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ProcessInstanceVariablesApi.prototype.createProcessInstanceVariables = function (processInstanceId, restVariables) {
        throwIfNotDefined(processInstanceId, 'processInstanceId');
        throwIfNotDefined(restVariables, 'restVariables');
        var postBody = restVariables;
        var pathParams = {
            'processInstanceId': processInstanceId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/variables', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ProcessInstanceVariablesApi.prototype.deleteProcessInstanceVariable = function (processInstanceId, variableName) {
        throwIfNotDefined(processInstanceId, 'processInstanceId');
        throwIfNotDefined(variableName, 'variableName');
        var postBody = null;
        var pathParams = {
            'processInstanceId': processInstanceId, 'variableName': variableName
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/variables/{variableName}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ProcessInstanceVariablesApi.prototype.getProcessInstanceVariable = function (processInstanceId, variableName) {
        throwIfNotDefined(processInstanceId, 'processInstanceId');
        throwIfNotDefined(variableName, 'variableName');
        var postBody = null;
        var pathParams = {
            'processInstanceId': processInstanceId, 'variableName': variableName
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/variables/{variableName}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RestVariable);
    };
    ProcessInstanceVariablesApi.prototype.getProcessInstanceVariables = function (processInstanceId) {
        throwIfNotDefined(processInstanceId, 'processInstanceId');
        var postBody = null;
        var pathParams = {
            'processInstanceId': processInstanceId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/variables', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    ProcessInstanceVariablesApi.prototype.updateProcessInstanceVariable = function (processInstanceId, variableName, restVariable) {
        throwIfNotDefined(processInstanceId, 'processInstanceId');
        throwIfNotDefined(variableName, 'variableName');
        throwIfNotDefined(restVariable, 'restVariable');
        var postBody = restVariable;
        var pathParams = {
            'processInstanceId': processInstanceId, 'variableName': variableName
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/process-instances/{processInstanceId}/variables/{variableName}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RestVariable);
    };
    return ProcessInstanceVariablesApi;
}(BaseApi));
export { ProcessInstanceVariablesApi };
