// @alfresco/js-api/esm5/src/api/activiti-rest-api/api/runtimeAppDeployments.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { AppDeploymentRepresentation } from '../model/appDeploymentRepresentation';
import { ResultListDataRepresentationAppDeploymentRepresentation } from '../model/resultListDataRepresentationAppDeploymentRepresentation';
import { BaseApi } from './base.api';
import { throwIfNotDefined } from '../../../assert';
var RuntimeAppDeploymentsApi = (function (_super) {
    __extends(RuntimeAppDeploymentsApi, _super);
    function RuntimeAppDeploymentsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RuntimeAppDeploymentsApi.prototype.deleteAppDeployment = function (appDeploymentId) {
        throwIfNotDefined(appDeploymentId, 'appDeploymentId');
        var postBody = null;
        var pathParams = {
            'appDeploymentId': appDeploymentId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/runtime-app-deployments/{appDeploymentId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    RuntimeAppDeploymentsApi.prototype.exportAppDefinition = function (deploymentId) {
        throwIfNotDefined(deploymentId, 'deploymentId');
        var postBody = null;
        var pathParams = {
            'deploymentId': deploymentId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/zip'];
        return this.apiClient.callApi('/api/enterprise/export-app-deployment/{deploymentId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    RuntimeAppDeploymentsApi.prototype.getAppDefinitions = function (opts) {
        opts = opts || {};
        var postBody = null;
        var pathParams = {};
        var queryParams = {
            'nameLike': opts['nameLike'],
            'tenantId': opts['tenantId'],
            'latest': opts['latest'],
            'start': opts['start'],
            'sort': opts['sort'],
            'order': opts['order'],
            'size': opts['size']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/runtime-app-deployments', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationAppDeploymentRepresentation);
    };
    RuntimeAppDeploymentsApi.prototype.getAppDeployment = function (appDeploymentId) {
        throwIfNotDefined(appDeploymentId, 'appDeploymentId');
        var postBody = null;
        var pathParams = {
            'appDeploymentId': appDeploymentId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/runtime-app-deployments/{appDeploymentId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, AppDeploymentRepresentation);
    };
    RuntimeAppDeploymentsApi.prototype.getRuntimeAppDeploymentByDeployment = function (opts) {
        opts = opts || {};
        var postBody = null;
        var pathParams = {};
        var queryParams = {
            'deploymentId': opts['deploymentId'],
            'dmnDeploymentId': opts['dmnDeploymentId']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/runtime-app-deployment', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, AppDeploymentRepresentation);
    };
    return RuntimeAppDeploymentsApi;
}(BaseApi));
export { RuntimeAppDeploymentsApi };
