// @alfresco/js-api/esm5/src/api/activiti-rest-api/api/userProfile.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { ImageUploadRepresentation } from '../model/imageUploadRepresentation';
import { UserRepresentation } from '../model/userRepresentation';
import { BaseApi } from './base.api';
import { throwIfNotDefined } from '../../../assert';
var UserProfileApi = (function (_super) {
    __extends(UserProfileApi, _super);
    function UserProfileApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UserProfileApi.prototype.changePassword = function (changePasswordRepresentation) {
        throwIfNotDefined(changePasswordRepresentation, 'changePasswordRepresentation');
        var postBody = changePasswordRepresentation;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/profile-password', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    UserProfileApi.prototype.getProfilePicture = function () {
        var postBody = null;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json', '*/*'];
        return this.apiClient.callApi('/api/enterprise/profile-picture', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    UserProfileApi.prototype.getProfilePictureUrl = function () {
        return this.apiClient.basePath + '/app/rest/admin/profile-picture';
    };
    UserProfileApi.prototype.getProfile = function () {
        var postBody = null;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/profile', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, UserRepresentation);
    };
    UserProfileApi.prototype.updateProfile = function (userRepresentation) {
        throwIfNotDefined(userRepresentation, 'userRepresentation');
        var postBody = userRepresentation;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/profile', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, UserRepresentation);
    };
    UserProfileApi.prototype.uploadProfilePicture = function (file) {
        throwIfNotDefined(file, 'file');
        var postBody = null;
        var pathParams = {};
        var queryParams = {};
        var headerParams = {};
        var formParams = {
            'file': file
        };
        var contentTypes = ['multipart/form-data'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/profile-picture', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ImageUploadRepresentation);
    };
    return UserProfileApi;
}(BaseApi));
export { UserProfileApi };
