// @alfresco/js-api/esm2015/src/api-legacy/activiti-rest-api/src/api/ContentApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { ContentApi as NewContentApi } from '../../../../api/activiti-rest-api/api/content.api';
import { ProcessInstancesApi } from '../../../../api/activiti-rest-api/api/processInstances.api';
export class ContentApi {
    init(alfrescoApi) {
        this.contentApi = new NewContentApi(alfrescoApi);
        this.processInstancesApi = new ProcessInstancesApi(alfrescoApi);
    }
    createRelatedContentOnProcessInstancePostBody(processInstanceId, relatedContent, opts) {
        return this.contentApi.createRelatedContentOnProcessInstance(processInstanceId, relatedContent, opts);
    }
    createRelatedContentOnProcessInstance(processInstanceId, file, opts) {
        return this.contentApi.createRelatedContentOnProcessInstance(processInstanceId, file, opts);
    }
    createRelatedContentOnTaskPostBody(taskId, relatedContent, opts) {
        return this.contentApi.createRelatedContentOnTask(taskId, relatedContent, opts);
    }
    createRelatedContentOnTask(taskId, file, opts) {
        return this.contentApi.createRelatedContentOnTask(taskId, file, opts);
    }
    createTemporaryRawRelatedContent(file) {
        return this.contentApi.createTemporaryRawRelatedContent(file);
    }
    createTemporaryRelatedContent(relatedContent) {
        return this.contentApi.createTemporaryRelatedContent(relatedContent);
    }
    deleteContent(contentId) {
        return this.contentApi.deleteContent(contentId);
    }
    getContent(contentId) {
        return this.contentApi.getContent(contentId);
    }
    getProcessInstanceContent(processInstanceId) {
        return this.processInstancesApi.getProcessInstanceContent(processInstanceId);
    }
    getRawContent(contentId) {
        return this.contentApi.getRawContent(contentId);
    }
    getContentPreview(contentId) {
        return this.contentApi.getRawContent(contentId, 'preview');
    }
    getRawContentUrl(contentId) {
        return this.contentApi.getRawContentUrl(contentId);
    }
    getContentThumbnail(contentId) {
        return this.contentApi.getRawContent(contentId, 'thumbnail');
    }
    getRelatedContentForProcessInstance(processInstanceId, isRelatedContent) {
        return this.contentApi.getRelatedContentForProcessInstance(processInstanceId, { isRelatedContent });
    }
    getRelatedContentForTask(taskId, isRelatedContent) {
        return this.contentApi.getRelatedContentForTask(taskId, { isRelatedContent });
    }
}
