// @alfresco/js-api/esm2015/src/api-legacy/activiti-rest-api/src/api/ProcessApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { ProcessInstancesApi } from '../../../../api/activiti-rest-api/api/processInstances.api';
import { ProcessDefinitionsApi } from '../../../../api/activiti-rest-api/api/processDefinitions.api';
export class ProcessApi {
    init(alfrescoApi) {
        this.processInstancesApi = new ProcessInstancesApi(alfrescoApi);
        this.processDefinitionsApi = new ProcessDefinitionsApi(alfrescoApi);
    }
    deleteProcessInstance(processInstanceId) {
        return this.processInstancesApi.deleteProcessInstance(processInstanceId);
    }
    filterProcessInstances(filterRequest) {
        return this.processInstancesApi.filterProcessInstances(filterRequest);
    }
    getProcessDefinitionStartForm(processDefinitionId) {
        return this.processDefinitionsApi.getProcessDefinitionStartForm(processDefinitionId);
    }
    getProcessDefinitions(opts) {
        return this.processDefinitionsApi.getProcessDefinitions(opts);
    }
    getProcessInstanceContent(processInstanceId) {
        return this.processInstancesApi.getProcessInstanceContent(processInstanceId);
    }
    getProcessInstanceStartForm(processInstanceId) {
        return this.processInstancesApi.getProcessInstanceStartForm(processInstanceId);
    }
    getProcessInstance(processInstanceId) {
        return this.processInstancesApi.getProcessInstance(processInstanceId);
    }
    getProcessInstances(processInstancesQuery) {
        return this.processInstancesApi.getProcessInstances(processInstancesQuery);
    }
    getProcessAuditJson(processInstanceId) {
        return this.processInstancesApi.getTaskAuditLog(processInstanceId);
    }
    getProcessAuditPdf(processInstanceId) {
        return this.processInstancesApi.getProcessAuditPdf(processInstanceId);
    }
    getRestFieldValues(processDefinitionId, field) {
        return this.processDefinitionsApi.getRestFieldValues(processDefinitionId, field);
    }
    getRestTableFieldValues(processDefinitionId, field, column) {
        return this.processDefinitionsApi.getRestTableFieldValues(processDefinitionId, field, column);
    }
    startNewProcessInstance(startRequest) {
        return this.processInstancesApi.startNewProcessInstance(startRequest);
    }
}
