// @alfresco/js-api/esm2015/src/api-legacy/content-rest-api/src/api/favoritesApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { FavoritesApi as NewFavoritesApi } from '../../../../api/content-rest-api/api/favorites.api';
export class FavoritesApi {
    init(alfrescoApi) {
        this.favoritesApi = new NewFavoritesApi(alfrescoApi);
    }
    addFavorite(personId, favoriteBody) {
        return this.favoritesApi.createFavorite(personId, favoriteBody);
    }
    getFavorite(personId, favoriteId, opts) {
        return this.favoritesApi.getFavorite(personId, favoriteId, opts);
    }
    getFavorites(personId, opts) {
        return this.favoritesApi.listFavorites(personId, opts);
    }
    removeFavoriteSite(personId, favoriteId) {
        return this.favoritesApi.deleteFavorite(personId, favoriteId);
    }
}
