// @alfresco/js-api/esm2015/src/api-legacy/content-rest-api/src/api/groupsApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { GroupsApi as NewGroupsApi } from '../../../../api/content-rest-api/api/groups.api';
export class GroupsApi {
    init(alfrescoApi) {
        this.groupsApi = new NewGroupsApi(alfrescoApi);
    }
    createGroup(groupBody, opts) {
        return this.groupsApi.createGroup(groupBody, opts);
    }
    getGroups(opts) {
        return this.groupsApi.listGroups(opts);
    }
    deleteGroup(groupId, opts) {
        return this.groupsApi.deleteGroup(groupId, opts);
    }
    getGroup(groupId, opts) {
        return this.groupsApi.getGroup(groupId, opts);
    }
    updateGroup(groupId, groupBody, opts) {
        return this.groupsApi.updateGroup(groupId, groupBody, opts);
    }
    getGroupMembers(groupId, opts) {
        return this.groupsApi.listGroupMemberships(groupId, opts);
    }
    addGroupMember(groupId, groupMemberBody, opts) {
        return this.groupsApi.createGroupMembership(groupId, groupMemberBody, opts);
    }
    deleteGroupMember(groupId, groupMemberId) {
        return this.groupsApi.deleteGroupMembership(groupId, groupMemberId);
    }
}
