// @alfresco/js-api/esm2015/src/api-legacy/content-rest-api/src/api/renditionsApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { RenditionsApi as NewRenditionsApi } from '../../../../api/content-rest-api/api/renditions.api';
import { SharedlinksApi } from '../../../../api/content-rest-api/api/sharedlinks.api';
export class RenditionsApi {
    init(alfrescoApi) {
        this.renditionsApi = new NewRenditionsApi(alfrescoApi);
        this.sharedlinksApi = new SharedlinksApi(alfrescoApi);
    }
    createRendition(nodeId, renditionBody) {
        return this.renditionsApi.createRendition(nodeId, renditionBody);
    }
    getRendition(nodeId, renditionId) {
        return this.renditionsApi.getRendition(nodeId, renditionId);
    }
    getRenditionContent(nodeId, renditionId, opts) {
        return this.renditionsApi.getRenditionContent(nodeId, renditionId, opts);
    }
    getRenditions(nodeId) {
        return this.renditionsApi.listRenditions(nodeId);
    }
    getSharedLinkRenditionContent(sharedId, renditionId, opts) {
        return this.sharedlinksApi.getSharedLinkRenditionContent(sharedId, renditionId, opts);
    }
    getSharedLinkRenditions(sharedId) {
        return this.sharedlinksApi.listSharedLinkRenditions(sharedId);
    }
    getSharedLinkRendition(sharedId, renditionId) {
        return this.sharedlinksApi.getSharedLinkRendition(sharedId, renditionId);
    }
}
