// @alfresco/js-api/esm2015/src/api-legacy/content-rest-api/src/api/sharedlinksApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { SharedlinksApi as NewSharedlinksApi } from '../../../../api/content-rest-api/api/sharedlinks.api';
export class SharedlinksApi {
    init(alfrescoApi) {
        this.sharedlinksApi = new NewSharedlinksApi(alfrescoApi);
    }
    addSharedLink(sharedLinkBody, opts) {
        return this.sharedlinksApi.createSharedLink(sharedLinkBody, opts);
    }
    deleteSharedLink(sharedId) {
        return this.sharedlinksApi.deleteSharedLink(sharedId);
    }
    emailSharedLink(sharedId, emailSharedLinkBody) {
        return this.sharedlinksApi.emailSharedLink(sharedId, emailSharedLinkBody);
    }
    findSharedLinks(opts) {
        return this.sharedlinksApi.listSharedLinks(opts);
    }
    getSharedLink(sharedId, opts) {
        return this.sharedlinksApi.getSharedLink(sharedId, opts);
    }
    getSharedLinkContent(sharedId, opts) {
        return this.sharedlinksApi.getSharedLinkContent(sharedId, opts);
    }
    getSharedLinkRenditionContent(sharedId, renditionId, opts) {
        return this.sharedlinksApi.getSharedLinkRenditionContent(sharedId, renditionId, opts);
    }
    getSharedLinkRenditions(sharedId) {
        return this.sharedlinksApi.listSharedLinkRenditions(sharedId);
    }
    getSharedLinkRendition(sharedId, renditionId) {
        return this.sharedlinksApi.getSharedLinkRendition(sharedId, renditionId);
    }
}
