// @alfresco/js-api/esm5/src/api-legacy/activiti-rest-api/src/index.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { AboutApi } from './api/AboutApi';
import { AlfrescoApiActiviti } from './api/AlfrescoApiActiviti';
import { AdminEndpointsApi } from './api/AdminEndpointsApi';
import { AdminGroupsApi } from './api/AdminGroupsApi';
import { AdminTenantsApi } from './api/AdminTenantsApi';
import { AdminUsersApi } from './api/AdminUsersApi';
import { AppsApi } from './api/AppsApi';
import { AppsDefinitionApi } from './api/AppsDefinitionApi';
import { AppsRuntimeApi } from './api/AppsRuntimeApi';
import { CommentsApi } from './api/CommentsApi';
import { ContentApi } from './api/ContentApi';
import { ContentRenditionApi } from './api/ContentRenditionApi';
import { EditorApi } from './api/EditorApi';
import { GroupsApi } from './api/GroupsApi';
import { IDMSyncApi } from './api/IDMSyncApi';
import { IntegrationAccountApi } from './api/IntegrationAccountApi';
import { IntegrationAlfrescoCloudApi } from './api/IntegrationAlfrescoCloudApi';
import { IntegrationAlfrescoOnPremiseApi } from './api/IntegrationAlfrescoOnPremiseApi';
import { IntegrationBoxApi } from './api/IntegrationBoxApi';
import { IntegrationDriveApi } from './api/IntegrationDriveApi';
import { ModelBpmnApi } from './api/ModelBpmnApi';
import { ModelJsonBpmnApi } from './api/ModelJsonBpmnApi';
import { ModelsApi } from './api/ModelsApi';
import { ModelsHistoryApi } from './api/ModelsHistoryApi';
import { ProcessApi } from './api/ProcessApi';
import { ProcessDefinitionsApi } from './api/ProcessDefinitionsApi';
import { ProcessDefinitionsFormApi } from './api/ProcessDefinitionsFormApi';
import { ProcessInstancesApi } from './api/ProcessInstancesApi';
import { ProcessInstancesInformationApi } from './api/ProcessInstancesInformationApi';
import { ProcessInstancesListingApi } from './api/ProcessInstancesListingApi';
import { ProcessInstanceVariablesApi } from './api/ProcessInstanceVariablesApi';
import { ProcessScopeApi } from './api/ProcessScopeApi';
import { ProfileApi } from './api/ProfileApi';
import { ScriptFileApi } from './api/ScriptFileApi';
import { SystemPropertiesApi } from './api/SystemPropertiesApi';
import { TaskApi } from './api/TaskApi';
import { TaskActionsApi } from './api/TaskActionsApi';
import { TaskCheckListApi } from './api/TaskCheckListApi';
import { TaskFormsApi } from './api/TaskFormsApi';
import { TemporaryApi } from './api/TemporaryApi';
import { UserApi } from './api/UserApi';
import { UserFiltersApi } from './api/UserFiltersApi';
import { UsersWorkflowApi } from './api/UsersWorkflowApi';
import { ReportApi } from './api/ReportApi';
import { AlfrescoApiActiviti as _AlfrescoApiActiviti } from './api/AlfrescoApiActiviti';
import { AboutApi as _AboutApi } from './api/AboutApi';
import { AdminEndpointsApi as _AdminEndpointsApi } from './api/AdminEndpointsApi';
import { AdminGroupsApi as _AdminGroupsApi } from './api/AdminGroupsApi';
import { AdminTenantsApi as _AdminTenantsApi } from './api/AdminTenantsApi';
import { AdminUsersApi as _AdminUsersApi } from './api/AdminUsersApi';
import { AppsApi as _AppsApi } from './api/AppsApi';
import { AppsDefinitionApi as _AppsDefinitionApi } from './api/AppsDefinitionApi';
import { AppsRuntimeApi as _AppsRuntimeApi } from './api/AppsRuntimeApi';
import { CommentsApi as _CommentsApi } from './api/CommentsApi';
import { ContentApi as _ContentApi } from './api/ContentApi';
import { ContentRenditionApi as _ContentRenditionApi } from './api/ContentRenditionApi';
import { EditorApi as _EditorApi } from './api/EditorApi';
import { GroupsApi as _GroupsApi } from './api/GroupsApi';
import { IDMSyncApi as _IDMSyncApi } from './api/IDMSyncApi';
import { IntegrationAccountApi as _IntegrationAccountApi } from './api/IntegrationAccountApi';
import { IntegrationAlfrescoCloudApi as _IntegrationAlfrescoCloudApi } from './api/IntegrationAlfrescoCloudApi';
import { IntegrationAlfrescoOnPremiseApi as _IntegrationAlfrescoOnPremiseApi } from './api/IntegrationAlfrescoOnPremiseApi';
import { IntegrationBoxApi as _IntegrationBoxApi } from './api/IntegrationBoxApi';
import { IntegrationDriveApi as _IntegrationDriveApi } from './api/IntegrationDriveApi';
import { ModelBpmnApi as _ModelBpmnApi } from './api/ModelBpmnApi';
import { ModelJsonBpmnApi as _ModelJsonBpmnApi } from './api/ModelJsonBpmnApi';
import { ModelsApi as _ModelsApi } from './api/ModelsApi';
import { ModelsHistoryApi as _ModelsHistoryApi } from './api/ModelsHistoryApi';
import { ProcessApi as _ProcessApi } from './api/ProcessApi';
import { ProcessDefinitionsApi as _ProcessDefinitionsApi } from './api/ProcessDefinitionsApi';
import { ProcessDefinitionsFormApi as _ProcessDefinitionsFormApi } from './api/ProcessDefinitionsFormApi';
import { ProcessInstancesApi as _ProcessInstancesApi } from './api/ProcessInstancesApi';
import { ProcessInstancesInformationApi as _ProcessInstancesInformationApi } from './api/ProcessInstancesInformationApi';
import { ProcessInstancesListingApi as _ProcessInstancesListingApi } from './api/ProcessInstancesListingApi';
import { ProcessInstanceVariablesApi as _ProcessInstanceVariablesApi } from './api/ProcessInstanceVariablesApi';
import { ProcessScopeApi as _ProcessScopeApi } from './api/ProcessScopeApi';
import { ProfileApi as _ProfileApi } from './api/ProfileApi';
import { ScriptFileApi as _ScriptFileApi } from './api/ScriptFileApi';
import { SystemPropertiesApi as _SystemPropertiesApi } from './api/SystemPropertiesApi';
import { TaskApi as _TaskApi } from './api/TaskApi';
import { TaskActionsApi as _TaskActionsApi } from './api/TaskActionsApi';
import { TaskCheckListApi as _TaskCheckListApi } from './api/TaskCheckListApi';
import { TaskFormsApi as _TaskFormsApi } from './api/TaskFormsApi';
import { TemporaryApi as _TemporaryApi } from './api/TemporaryApi';
import { UserApi as _UserApi } from './api/UserApi';
import { UserFiltersApi as _UserFiltersApi } from './api/UserFiltersApi';
import { UsersWorkflowApi as _UsersWorkflowApi } from './api/UsersWorkflowApi';
import { ReportApi as _ReportApi } from './api/ReportApi';
export var APS_LEGACY_APIS = {
    AlfrescoApi: AlfrescoApiActiviti,
    AboutApi: AboutApi,
    AdminEndpointsApi: AdminEndpointsApi,
    AdminGroupsApi: AdminGroupsApi,
    AdminTenantsApi: AdminTenantsApi,
    AdminUsersApi: AdminUsersApi,
    AppsApi: AppsApi,
    AppsDefinitionApi: AppsDefinitionApi,
    AppsRuntimeApi: AppsRuntimeApi,
    CommentsApi: CommentsApi,
    ContentApi: ContentApi,
    ContentRenditionApi: ContentRenditionApi,
    EditorApi: EditorApi,
    GroupsApi: GroupsApi,
    IDMSyncApi: IDMSyncApi,
    IntegrationAccountApi: IntegrationAccountApi,
    IntegrationAlfrescoCloudApi: IntegrationAlfrescoCloudApi,
    IntegrationAlfrescoOnPremiseApi: IntegrationAlfrescoOnPremiseApi,
    IntegrationBoxApi: IntegrationBoxApi,
    IntegrationDriveApi: IntegrationDriveApi,
    ModelBpmnApi: ModelBpmnApi,
    ModelJsonBpmnApi: ModelJsonBpmnApi,
    ModelsApi: ModelsApi,
    ModelsHistoryApi: ModelsHistoryApi,
    ProcessApi: ProcessApi,
    ProcessDefinitionsApi: ProcessDefinitionsApi,
    ProcessDefinitionsFormApi: ProcessDefinitionsFormApi,
    ProcessInstancesApi: ProcessInstancesApi,
    ProcessInstancesInformationApi: ProcessInstancesInformationApi,
    ProcessInstancesListingApi: ProcessInstancesListingApi,
    ProcessInstanceVariablesApi: ProcessInstanceVariablesApi,
    ProcessScopeApi: ProcessScopeApi,
    ProfileApi: ProfileApi,
    ScriptFileApi: ScriptFileApi,
    SystemPropertiesApi: SystemPropertiesApi,
    TaskApi: TaskApi,
    TaskActionsApi: TaskActionsApi,
    TaskCheckListApi: TaskCheckListApi,
    TaskFormsApi: TaskFormsApi,
    TemporaryApi: TemporaryApi,
    UserApi: UserApi,
    UserFiltersApi: UserFiltersApi,
    UsersWorkflowApi: UsersWorkflowApi,
    ReportApi: ReportApi
};
export var Activiti;
(function (Activiti) {
    var AlfrescoApi = (function (_super) {
        __extends(AlfrescoApi, _super);
        function AlfrescoApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AlfrescoApi;
    }(_AlfrescoApiActiviti));
    Activiti.AlfrescoApi = AlfrescoApi;
    var AboutApi = (function (_super) {
        __extends(AboutApi, _super);
        function AboutApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AboutApi;
    }(_AboutApi));
    Activiti.AboutApi = AboutApi;
    var AdminEndpointsApi = (function (_super) {
        __extends(AdminEndpointsApi, _super);
        function AdminEndpointsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AdminEndpointsApi;
    }(_AdminEndpointsApi));
    Activiti.AdminEndpointsApi = AdminEndpointsApi;
    var AdminGroupsApi = (function (_super) {
        __extends(AdminGroupsApi, _super);
        function AdminGroupsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AdminGroupsApi;
    }(_AdminGroupsApi));
    Activiti.AdminGroupsApi = AdminGroupsApi;
    var AdminTenantsApi = (function (_super) {
        __extends(AdminTenantsApi, _super);
        function AdminTenantsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AdminTenantsApi;
    }(_AdminTenantsApi));
    Activiti.AdminTenantsApi = AdminTenantsApi;
    var AdminUsersApi = (function (_super) {
        __extends(AdminUsersApi, _super);
        function AdminUsersApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AdminUsersApi;
    }(_AdminUsersApi));
    Activiti.AdminUsersApi = AdminUsersApi;
    var AppsApi = (function (_super) {
        __extends(AppsApi, _super);
        function AppsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AppsApi;
    }(_AppsApi));
    Activiti.AppsApi = AppsApi;
    var AppsDefinitionApi = (function (_super) {
        __extends(AppsDefinitionApi, _super);
        function AppsDefinitionApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AppsDefinitionApi;
    }(_AppsDefinitionApi));
    Activiti.AppsDefinitionApi = AppsDefinitionApi;
    var AppsRuntimeApi = (function (_super) {
        __extends(AppsRuntimeApi, _super);
        function AppsRuntimeApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return AppsRuntimeApi;
    }(_AppsRuntimeApi));
    Activiti.AppsRuntimeApi = AppsRuntimeApi;
    var CommentsApi = (function (_super) {
        __extends(CommentsApi, _super);
        function CommentsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return CommentsApi;
    }(_CommentsApi));
    Activiti.CommentsApi = CommentsApi;
    var ContentApi = (function (_super) {
        __extends(ContentApi, _super);
        function ContentApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ContentApi;
    }(_ContentApi));
    Activiti.ContentApi = ContentApi;
    var ContentRenditionApi = (function (_super) {
        __extends(ContentRenditionApi, _super);
        function ContentRenditionApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ContentRenditionApi;
    }(_ContentRenditionApi));
    Activiti.ContentRenditionApi = ContentRenditionApi;
    var EditorApi = (function (_super) {
        __extends(EditorApi, _super);
        function EditorApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return EditorApi;
    }(_EditorApi));
    Activiti.EditorApi = EditorApi;
    var GroupsApi = (function (_super) {
        __extends(GroupsApi, _super);
        function GroupsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return GroupsApi;
    }(_GroupsApi));
    Activiti.GroupsApi = GroupsApi;
    var IDMSyncApi = (function (_super) {
        __extends(IDMSyncApi, _super);
        function IDMSyncApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return IDMSyncApi;
    }(_IDMSyncApi));
    Activiti.IDMSyncApi = IDMSyncApi;
    var IntegrationAccountApi = (function (_super) {
        __extends(IntegrationAccountApi, _super);
        function IntegrationAccountApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return IntegrationAccountApi;
    }(_IntegrationAccountApi));
    Activiti.IntegrationAccountApi = IntegrationAccountApi;
    var IntegrationAlfrescoCloudApi = (function (_super) {
        __extends(IntegrationAlfrescoCloudApi, _super);
        function IntegrationAlfrescoCloudApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return IntegrationAlfrescoCloudApi;
    }(_IntegrationAlfrescoCloudApi));
    Activiti.IntegrationAlfrescoCloudApi = IntegrationAlfrescoCloudApi;
    var IntegrationAlfrescoOnPremiseApi = (function (_super) {
        __extends(IntegrationAlfrescoOnPremiseApi, _super);
        function IntegrationAlfrescoOnPremiseApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return IntegrationAlfrescoOnPremiseApi;
    }(_IntegrationAlfrescoOnPremiseApi));
    Activiti.IntegrationAlfrescoOnPremiseApi = IntegrationAlfrescoOnPremiseApi;
    var IntegrationBoxApi = (function (_super) {
        __extends(IntegrationBoxApi, _super);
        function IntegrationBoxApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return IntegrationBoxApi;
    }(_IntegrationBoxApi));
    Activiti.IntegrationBoxApi = IntegrationBoxApi;
    var IntegrationDriveApi = (function (_super) {
        __extends(IntegrationDriveApi, _super);
        function IntegrationDriveApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return IntegrationDriveApi;
    }(_IntegrationDriveApi));
    Activiti.IntegrationDriveApi = IntegrationDriveApi;
    var ModelBpmnApi = (function (_super) {
        __extends(ModelBpmnApi, _super);
        function ModelBpmnApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ModelBpmnApi;
    }(_ModelBpmnApi));
    Activiti.ModelBpmnApi = ModelBpmnApi;
    var ModelJsonBpmnApi = (function (_super) {
        __extends(ModelJsonBpmnApi, _super);
        function ModelJsonBpmnApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ModelJsonBpmnApi;
    }(_ModelJsonBpmnApi));
    Activiti.ModelJsonBpmnApi = ModelJsonBpmnApi;
    var ModelsApi = (function (_super) {
        __extends(ModelsApi, _super);
        function ModelsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ModelsApi;
    }(_ModelsApi));
    Activiti.ModelsApi = ModelsApi;
    var ModelsHistoryApi = (function (_super) {
        __extends(ModelsHistoryApi, _super);
        function ModelsHistoryApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ModelsHistoryApi;
    }(_ModelsHistoryApi));
    Activiti.ModelsHistoryApi = ModelsHistoryApi;
    var ProcessApi = (function (_super) {
        __extends(ProcessApi, _super);
        function ProcessApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProcessApi;
    }(_ProcessApi));
    Activiti.ProcessApi = ProcessApi;
    var ProcessDefinitionsApi = (function (_super) {
        __extends(ProcessDefinitionsApi, _super);
        function ProcessDefinitionsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProcessDefinitionsApi;
    }(_ProcessDefinitionsApi));
    Activiti.ProcessDefinitionsApi = ProcessDefinitionsApi;
    var ProcessDefinitionsFormApi = (function (_super) {
        __extends(ProcessDefinitionsFormApi, _super);
        function ProcessDefinitionsFormApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProcessDefinitionsFormApi;
    }(_ProcessDefinitionsFormApi));
    Activiti.ProcessDefinitionsFormApi = ProcessDefinitionsFormApi;
    var ProcessInstancesApi = (function (_super) {
        __extends(ProcessInstancesApi, _super);
        function ProcessInstancesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProcessInstancesApi;
    }(_ProcessInstancesApi));
    Activiti.ProcessInstancesApi = ProcessInstancesApi;
    var ProcessInstancesInformationApi = (function (_super) {
        __extends(ProcessInstancesInformationApi, _super);
        function ProcessInstancesInformationApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProcessInstancesInformationApi;
    }(_ProcessInstancesInformationApi));
    Activiti.ProcessInstancesInformationApi = ProcessInstancesInformationApi;
    var ProcessInstancesListingApi = (function (_super) {
        __extends(ProcessInstancesListingApi, _super);
        function ProcessInstancesListingApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProcessInstancesListingApi;
    }(_ProcessInstancesListingApi));
    Activiti.ProcessInstancesListingApi = ProcessInstancesListingApi;
    var ProcessInstanceVariablesApi = (function (_super) {
        __extends(ProcessInstanceVariablesApi, _super);
        function ProcessInstanceVariablesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProcessInstanceVariablesApi;
    }(_ProcessInstanceVariablesApi));
    Activiti.ProcessInstanceVariablesApi = ProcessInstanceVariablesApi;
    var ProcessScopeApi = (function (_super) {
        __extends(ProcessScopeApi, _super);
        function ProcessScopeApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProcessScopeApi;
    }(_ProcessScopeApi));
    Activiti.ProcessScopeApi = ProcessScopeApi;
    var ProfileApi = (function (_super) {
        __extends(ProfileApi, _super);
        function ProfileApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ProfileApi;
    }(_ProfileApi));
    Activiti.ProfileApi = ProfileApi;
    var ScriptFileApi = (function (_super) {
        __extends(ScriptFileApi, _super);
        function ScriptFileApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ScriptFileApi;
    }(_ScriptFileApi));
    Activiti.ScriptFileApi = ScriptFileApi;
    var SystemPropertiesApi = (function (_super) {
        __extends(SystemPropertiesApi, _super);
        function SystemPropertiesApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return SystemPropertiesApi;
    }(_SystemPropertiesApi));
    Activiti.SystemPropertiesApi = SystemPropertiesApi;
    var TaskApi = (function (_super) {
        __extends(TaskApi, _super);
        function TaskApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return TaskApi;
    }(_TaskApi));
    Activiti.TaskApi = TaskApi;
    var TaskActionsApi = (function (_super) {
        __extends(TaskActionsApi, _super);
        function TaskActionsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return TaskActionsApi;
    }(_TaskActionsApi));
    Activiti.TaskActionsApi = TaskActionsApi;
    var TaskCheckListApi = (function (_super) {
        __extends(TaskCheckListApi, _super);
        function TaskCheckListApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return TaskCheckListApi;
    }(_TaskCheckListApi));
    Activiti.TaskCheckListApi = TaskCheckListApi;
    var TaskFormsApi = (function (_super) {
        __extends(TaskFormsApi, _super);
        function TaskFormsApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return TaskFormsApi;
    }(_TaskFormsApi));
    Activiti.TaskFormsApi = TaskFormsApi;
    var TemporaryApi = (function (_super) {
        __extends(TemporaryApi, _super);
        function TemporaryApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return TemporaryApi;
    }(_TemporaryApi));
    Activiti.TemporaryApi = TemporaryApi;
    var UserApi = (function (_super) {
        __extends(UserApi, _super);
        function UserApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return UserApi;
    }(_UserApi));
    Activiti.UserApi = UserApi;
    var UserFiltersApi = (function (_super) {
        __extends(UserFiltersApi, _super);
        function UserFiltersApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return UserFiltersApi;
    }(_UserFiltersApi));
    Activiti.UserFiltersApi = UserFiltersApi;
    var UsersWorkflowApi = (function (_super) {
        __extends(UsersWorkflowApi, _super);
        function UsersWorkflowApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return UsersWorkflowApi;
    }(_UsersWorkflowApi));
    Activiti.UsersWorkflowApi = UsersWorkflowApi;
    var ReportApi = (function (_super) {
        __extends(ReportApi, _super);
        function ReportApi() {
            return _super !== null && _super.apply(this, arguments) || this;
        }
        return ReportApi;
    }(_ReportApi));
    Activiti.ReportApi = ReportApi;
})(Activiti || (Activiti = {}));
