// @alfresco/js-api/src/api-legacy/content-rest-api/src/index.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.Core = exports.CONTENT_LEGACY_APIS = void 0;
const changesApi_1 = require("./api/changesApi");
const childAssociationsApi_1 = require("./api/childAssociationsApi");
const commentsApi_1 = require("./api/commentsApi");
const favoritesApi_1 = require("./api/favoritesApi");
const nodesApi_1 = require("./api/nodesApi");
const networksApi_1 = require("./api/networksApi");
const ratingsApi_1 = require("./api/ratingsApi");
const peopleApi_1 = require("./api/peopleApi");
const renditionsApi_1 = require("./api/renditionsApi");
const queriesApi_1 = require("./api/queriesApi");
const sharedlinksApi_1 = require("./api/sharedlinksApi");
const sitesApi_1 = require("./api/sitesApi");
const tagsApi_1 = require("./api/tagsApi");
const webscriptApi_1 = require("./api/webscriptApi");
const downloadsApi_1 = require("./api/downloadsApi");
const versionsApi_1 = require("./api/versionsApi");
const classesApi_1 = require("./api/classesApi");
const groupsApi_1 = require("./api/groupsApi");
const customModelApi_1 = require("./api/customModelApi");
const nodesApi_2 = require("./api/nodesApi");
const changesApi_2 = require("./api/changesApi");
const childAssociationsApi_2 = require("./api/childAssociationsApi");
const commentsApi_2 = require("./api/commentsApi");
const favoritesApi_2 = require("./api/favoritesApi");
const networksApi_2 = require("./api/networksApi");
const ratingsApi_2 = require("./api/ratingsApi");
const peopleApi_2 = require("./api/peopleApi");
const renditionsApi_2 = require("./api/renditionsApi");
const queriesApi_2 = require("./api/queriesApi");
const sharedlinksApi_2 = require("./api/sharedlinksApi");
const sitesApi_2 = require("./api/sitesApi");
const tagsApi_2 = require("./api/tagsApi");
const webscriptApi_2 = require("./api/webscriptApi");
const downloadsApi_2 = require("./api/downloadsApi");
const versionsApi_2 = require("./api/versionsApi");
const classesApi_2 = require("./api/classesApi");
const groupsApi_2 = require("./api/groupsApi");
const customModelApi_2 = require("./api/customModelApi");
exports.CONTENT_LEGACY_APIS = {
    ChangesApi: changesApi_1.ChangesApi,
    ChildAssociationsApi: childAssociationsApi_1.ChildAssociationsApi,
    FavoritesApi: favoritesApi_1.FavoritesApi,
    NetworksApi: networksApi_1.NetworksApi,
    NodesApi: nodesApi_1.NodesApi,
    CommentsApi: commentsApi_1.CommentsApi,
    PeopleApi: peopleApi_1.PeopleApi,
    RatingsApi: ratingsApi_1.RatingsApi,
    RenditionsApi: renditionsApi_1.RenditionsApi,
    QueriesApi: queriesApi_1.QueriesApi,
    SharedlinksApi: sharedlinksApi_1.SharedlinksApi,
    SitesApi: sitesApi_1.SitesApi,
    TagsApi: tagsApi_1.TagsApi,
    WebscriptApi: webscriptApi_1.WebscriptApi,
    DownloadsApi: downloadsApi_1.DownloadsApi,
    VersionsApi: versionsApi_1.VersionsApi,
    ClassesApi: classesApi_1.ClassesApi,
    GroupsApi: groupsApi_1.GroupsApi,
    CustomModelApi: customModelApi_1.CustomModelApi
};
var Core;
(function (Core) {
    class ChangesApi extends changesApi_2.ChangesApi {
    }
    Core.ChangesApi = ChangesApi;
    class ChildAssociationsApi extends childAssociationsApi_2.ChildAssociationsApi {
    }
    Core.ChildAssociationsApi = ChildAssociationsApi;
    class FavoritesApi extends favoritesApi_2.FavoritesApi {
    }
    Core.FavoritesApi = FavoritesApi;
    class NetworksApi extends networksApi_2.NetworksApi {
    }
    Core.NetworksApi = NetworksApi;
    class NodesApi extends nodesApi_2.NodesApi {
    }
    Core.NodesApi = NodesApi;
    class CommentsApi extends commentsApi_2.CommentsApi {
    }
    Core.CommentsApi = CommentsApi;
    class PeopleApi extends peopleApi_2.PeopleApi {
    }
    Core.PeopleApi = PeopleApi;
    class RatingsApi extends ratingsApi_2.RatingsApi {
    }
    Core.RatingsApi = RatingsApi;
    class RenditionsApi extends renditionsApi_2.RenditionsApi {
    }
    Core.RenditionsApi = RenditionsApi;
    class QueriesApi extends queriesApi_2.QueriesApi {
    }
    Core.QueriesApi = QueriesApi;
    class SharedlinksApi extends sharedlinksApi_2.SharedlinksApi {
    }
    Core.SharedlinksApi = SharedlinksApi;
    class SitesApi extends sitesApi_2.SitesApi {
    }
    Core.SitesApi = SitesApi;
    class TagsApi extends tagsApi_2.TagsApi {
    }
    Core.TagsApi = TagsApi;
    class WebscriptApi extends webscriptApi_2.WebscriptApi {
    }
    Core.WebscriptApi = WebscriptApi;
    class DownloadsApi extends downloadsApi_2.DownloadsApi {
    }
    Core.DownloadsApi = DownloadsApi;
    class VersionsApi extends versionsApi_2.VersionsApi {
    }
    Core.VersionsApi = VersionsApi;
    class ClassesApi extends classesApi_2.ClassesApi {
    }
    Core.ClassesApi = ClassesApi;
    class GroupsApi extends groupsApi_2.GroupsApi {
    }
    Core.GroupsApi = GroupsApi;
    class CustomModelApi extends customModelApi_2.CustomModelApi {
    }
    Core.CustomModelApi = CustomModelApi;
})(Core = exports.Core || (exports.Core = {}));
