// @alfresco/js-api/esm2015/src/api-legacy/legacy.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
export { ChildAssociationBody as AssocChildBody } from '../api/content-rest-api/model/childAssociationBody';
export { AssociationBody as AssocTargetBody } from '../api/content-rest-api/model/associationBody';
export { FavoriteBodyCreate as FavoriteBody } from '../api/content-rest-api/model/favoriteBodyCreate';
export { SearchRequest as QueryBody } from '../api/search-rest-api/model/searchRequest';
export { PathInfo as PathInfoEntity } from '../api/content-rest-api/model/pathInfo';
export { NodeEntry as MinimalNodeEntity } from '../api/content-rest-api/model/nodeEntry';
export { DeletedNodeEntry as DeletedNodeEntity } from '../api/content-rest-api/model/deletedNodeEntry';
export { Node as MinimalNode } from '../api/content-rest-api/model/node';
export { Node as MinimalNodeEntryEntity } from '../api/content-rest-api/model/node';
export { PathElement as PathElementEntity } from '../api/content-rest-api/model/pathElement';
export { SiteBodyCreate as SiteBody } from '../api/content-rest-api/model/siteBodyCreate';
export { GroupPaging as GroupsPaging } from '../api/content-rest-api/model/groupPaging';
export { ProcessInstanceQueryRepresentation as ProcessFilterRequestRepresentation } from '../api/activiti-rest-api/model/processInstanceQueryRepresentation';
export { TaskQueryRepresentation as TaskQueryRequestRepresentation } from '../api/activiti-rest-api/model/taskQueryRepresentation';
export { ContentClient as EcmClient } from '../contentClient';
export { ProcessClient as BpmClient } from '../processClient';
export { SiteMembershipBodyCreate as SiteMemberBody } from '../api/content-rest-api/model/siteMembershipBodyCreate';
export { SiteMembershipBodyUpdate as SiteMemberRoleBody } from '../api/content-rest-api/model/siteMembershipBodyUpdate';
export { SharedLinkBodyCreate as SharedLinkBody } from '../api/content-rest-api/model/sharedLinkBodyCreate';
export { SharedLinkBodyEmail as EmailSharedLinkBody } from '../api/content-rest-api/model/sharedLinkBodyEmail';
export { RenditionBodyCreate as RenditionBody } from '../api/content-rest-api/model/renditionBodyCreate';
export { SiteMembershipRequestBodyCreate as SiteMembershipRequestBody } from '../api/content-rest-api/model/siteMembershipRequestBodyCreate';
export { FavoriteSiteBodyCreate as FavoriteSiteBody } from '../api/content-rest-api/model/favoriteSiteBodyCreate';
export { SiteMembershipRequestBodyUpdate as SiteMembershipBody } from '../api/content-rest-api/model/siteMembershipRequestBodyUpdate';
export { NodeBodyCreate as NodeBody } from '../api/content-rest-api/model/nodeBodyCreate';
export { NodeBodyCopy as CopyBody } from '../api/content-rest-api/model/nodeBodyCopy';
export { NodeBodyMove as MoveBody } from '../api/content-rest-api/model/nodeBodyMove';
export { GroupBodyCreate as GroupBody } from '../api/content-rest-api/model/groupBodyCreate';
