// @alfresco/js-api/esm2015/src/api/activiti-rest-api/api/adminEndpoints.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { EndpointBasicAuthRepresentation } from '../model/endpointBasicAuthRepresentation';
import { EndpointConfigurationRepresentation } from '../model/endpointConfigurationRepresentation';
import { BaseApi } from './base.api';
import { throwIfNotDefined } from '../../../assert';
export class AdminEndpointsApi extends BaseApi {
    createBasicAuthConfiguration(createRepresentation) {
        throwIfNotDefined(createRepresentation, 'createRepresentation');
        let postBody = createRepresentation;
        let pathParams = {};
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointBasicAuthRepresentation);
    }
    createEndpointConfiguration(representation) {
        throwIfNotDefined(representation, 'representation');
        let postBody = representation;
        let pathParams = {};
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointConfigurationRepresentation);
    }
    getBasicAuthConfiguration(basicAuthId, tenantId) {
        throwIfNotDefined(basicAuthId, 'basicAuthId');
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {
            'basicAuthId': basicAuthId
        };
        let queryParams = {
            'tenantId': tenantId
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths/{basicAuthId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointBasicAuthRepresentation);
    }
    getBasicAuthConfigurations(tenantId) {
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'tenantId': tenantId
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointBasicAuthRepresentation);
    }
    getEndpointConfiguration(endpointConfigurationId, tenantId) {
        throwIfNotDefined(endpointConfigurationId, 'endpointConfigurationId');
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {
            'endpointConfigurationId': endpointConfigurationId
        };
        let queryParams = {
            'tenantId': tenantId
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints/{endpointConfigurationId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointConfigurationRepresentation);
    }
    getEndpointConfigurations(tenantId) {
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'tenantId': tenantId
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointConfigurationRepresentation);
    }
    removeBasicAuthConfiguration(basicAuthId, tenantId) {
        throwIfNotDefined(basicAuthId, 'basicAuthId');
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {
            'basicAuthId': basicAuthId
        };
        let queryParams = {
            'tenantId': tenantId
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths/{basicAuthId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    removeEndpointConfiguration(endpointConfigurationId, tenantId) {
        throwIfNotDefined(endpointConfigurationId, 'endpointConfigurationId');
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {
            'endpointConfigurationId': endpointConfigurationId
        };
        let queryParams = {
            'tenantId': tenantId
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints/{endpointConfigurationId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    updateBasicAuthConfiguration(basicAuthId, createRepresentation) {
        throwIfNotDefined(basicAuthId, 'basicAuthId');
        throwIfNotDefined(createRepresentation, 'createRepresentation');
        let postBody = createRepresentation;
        let pathParams = {
            'basicAuthId': basicAuthId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/basic-auths/{basicAuthId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointBasicAuthRepresentation);
    }
    updateEndpointConfiguration(endpointConfigurationId, representation) {
        throwIfNotDefined(endpointConfigurationId, 'endpointConfigurationId');
        throwIfNotDefined(representation, 'representation');
        let postBody = representation;
        let pathParams = {
            'endpointConfigurationId': endpointConfigurationId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/endpoints/{endpointConfigurationId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, EndpointConfigurationRepresentation);
    }
}
