// @alfresco/js-api/esm2015/src/api/activiti-rest-api/api/adminGroups.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { AbstractGroupRepresentation } from '../model/abstractGroupRepresentation';
import { GroupRepresentation } from '../model/groupRepresentation';
import { LightGroupRepresentation } from '../model/lightGroupRepresentation';
import { ResultListDataRepresentationLightUserRepresentation } from '../model/resultListDataRepresentationLightUserRepresentation';
import { BaseApi } from './base.api';
import { throwIfNotDefined } from '../../../assert';
export class AdminGroupsApi extends BaseApi {
    activate(groupId) {
        throwIfNotDefined(groupId, 'groupId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/action/activate', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    addAllUsersToGroup(groupId) {
        throwIfNotDefined(groupId, 'groupId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/add-all-users', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    addGroupCapabilities(groupId, addGroupCapabilitiesRepresentation) {
        throwIfNotDefined(groupId, 'groupId');
        throwIfNotDefined(addGroupCapabilitiesRepresentation, 'addGroupCapabilitiesRepresentation');
        let postBody = addGroupCapabilitiesRepresentation;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/capabilities', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    addGroupMember(groupId, userId) {
        throwIfNotDefined(groupId, 'groupId');
        throwIfNotDefined(userId, 'userId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId, 'userId': userId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/members/{userId}', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    addRelatedGroup(groupId, relatedGroupId, type) {
        throwIfNotDefined(groupId, 'groupId');
        throwIfNotDefined(relatedGroupId, 'relatedGroupId');
        throwIfNotDefined(type, 'type');
        let postBody = null;
        let pathParams = {
            'groupId': groupId, 'relatedGroupId': relatedGroupId
        };
        let queryParams = {
            'type': type
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/related-groups/{relatedGroupId}', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    createNewGroup(groupRepresentation) {
        throwIfNotDefined(groupRepresentation, 'groupRepresentation');
        let postBody = groupRepresentation;
        let pathParams = {};
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupRepresentation);
    }
    deleteGroupCapability(groupId, groupCapabilityId) {
        throwIfNotDefined(groupId, 'groupId');
        throwIfNotDefined(groupCapabilityId, 'groupCapabilityId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId, 'groupCapabilityId': groupCapabilityId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/capabilities/{groupCapabilityId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteGroupMember(groupId, userId) {
        throwIfNotDefined(groupId, 'groupId');
        throwIfNotDefined(userId, 'userId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId, 'userId': userId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/members/{userId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteGroup(groupId) {
        throwIfNotDefined(groupId, 'groupId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    deleteRelatedGroup(groupId, relatedGroupId) {
        throwIfNotDefined(groupId, 'groupId');
        throwIfNotDefined(relatedGroupId, 'relatedGroupId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId, 'relatedGroupId': relatedGroupId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/related-groups/{relatedGroupId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getCapabilities(groupId) {
        throwIfNotDefined(groupId, 'groupId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/potential-capabilities', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getGroupUsers(groupId, opts) {
        throwIfNotDefined(groupId, 'groupId');
        opts = opts || {};
        let postBody = null;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {
            'filter': opts['filter'],
            'page': opts['page'],
            'pageSize': opts['pageSize']
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/users', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ResultListDataRepresentationLightUserRepresentation);
    }
    getGroup(groupId, opts) {
        throwIfNotDefined(groupId, 'groupId');
        opts = opts || {};
        let postBody = null;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {
            'includeAllUsers': opts['includeAllUsers'],
            'summary': opts['summary']
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, AbstractGroupRepresentation);
    }
    getGroups(opts) {
        opts = opts || {};
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'tenantId': opts['tenantId'],
            'functional': opts['functional'],
            'summary': opts['summary']
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, LightGroupRepresentation);
    }
    getRelatedGroups(groupId) {
        throwIfNotDefined(groupId, 'groupId');
        let postBody = null;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}/related-groups', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, LightGroupRepresentation);
    }
    updateGroup(groupId, groupRepresentation) {
        throwIfNotDefined(groupId, 'groupId');
        throwIfNotDefined(groupRepresentation, 'groupRepresentation');
        let postBody = groupRepresentation;
        let pathParams = {
            'groupId': groupId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/groups/{groupId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, GroupRepresentation);
    }
}
