// @alfresco/js-api/esm2015/src/api/activiti-rest-api/api/adminTenants.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { ImageUploadRepresentation } from '../model/imageUploadRepresentation';
import { LightTenantRepresentation } from '../model/lightTenantRepresentation';
import { TenantEvent } from '../model/tenantEvent';
import { TenantRepresentation } from '../model/tenantRepresentation';
import { BaseApi } from './base.api';
import { throwIfNotDefined } from '../../../assert';
export class AdminTenantsApi extends BaseApi {
    createTenant(createTenantRepresentation) {
        throwIfNotDefined(createTenantRepresentation, 'groupId');
        let postBody = createTenantRepresentation;
        let pathParams = {};
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/tenants', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, LightTenantRepresentation);
    }
    deleteTenant(tenantId) {
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {
            'tenantId': tenantId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/tenants/{tenantId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getTenantEvents(tenantId) {
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {
            'tenantId': tenantId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/tenants/{tenantId}/events', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, TenantEvent);
    }
    getTenantLogo(tenantId) {
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {
            'tenantId': tenantId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/tenants/{tenantId}/logo', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getTenant(tenantId) {
        throwIfNotDefined(tenantId, 'tenantId');
        let postBody = null;
        let pathParams = {
            'tenantId': tenantId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/tenants/{tenantId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, TenantRepresentation);
    }
    getTenants() {
        let postBody = null;
        let pathParams = {};
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/tenants', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, LightTenantRepresentation);
    }
    update(tenantId, createTenantRepresentation) {
        throwIfNotDefined(tenantId, 'tenantId');
        throwIfNotDefined(createTenantRepresentation, 'createTenantRepresentation');
        let postBody = createTenantRepresentation;
        let pathParams = {
            'tenantId': tenantId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/tenants/{tenantId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, TenantRepresentation);
    }
    uploadTenantLogo(tenantId, file) {
        throwIfNotDefined(tenantId, 'tenantId');
        throwIfNotDefined(file, 'file');
        let postBody = null;
        let pathParams = {
            'tenantId': tenantId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {
            'file': file
        };
        let contentTypes = ['multipart/form-data'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/api/enterprise/admin/tenants/{tenantId}/logo', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ImageUploadRepresentation);
    }
}
