// @alfresco/js-api/src/api/gs-classification-rest-api/api/index.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.GS_CLASSIFICATION_APIS = void 0;
const tslib_1 = require("tslib");
(0, tslib_1.__exportStar)(require("./classificationGuides.api"), exports);
const classificationGuides_api_1 = require("./classificationGuides.api");
(0, tslib_1.__exportStar)(require("./classificationReasons.api"), exports);
const classificationReasons_api_1 = require("./classificationReasons.api");
(0, tslib_1.__exportStar)(require("./declassificationExemptions.api"), exports);
const declassificationExemptions_api_1 = require("./declassificationExemptions.api");
(0, tslib_1.__exportStar)(require("./defaultClassificationValues.api"), exports);
const defaultClassificationValues_api_1 = require("./defaultClassificationValues.api");
(0, tslib_1.__exportStar)(require("./securityControlSettings.api"), exports);
const securityControlSettings_api_1 = require("./securityControlSettings.api");
exports.GS_CLASSIFICATION_APIS = {
    ClassificationGuidesApi: classificationGuides_api_1.ClassificationGuidesApi,
    ClassificationReasonsApi: classificationReasons_api_1.ClassificationReasonsApi,
    DeclassificationExemptionsApi: declassificationExemptions_api_1.DeclassificationExemptionsApi,
    DefaultClassificationValuesApi: defaultClassificationValues_api_1.DefaultClassificationValuesApi,
    SecurityControlSettingsApi: securityControlSettings_api_1.SecurityControlSettingsApi
};
