// @alfresco/js-api/esm2015/src/alfrescoApiCompatibility.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { AlfrescoApi } from './alfrescoApi';
import { CONTENT_LEGACY_APIS } from './api-legacy/content-rest-api/src';
import { APS_LEGACY_APIS } from './api-legacy/activiti-rest-api/src';
import { DISCOVERY_APIS } from './api/discovery-rest-api/api';
import { GS_CLASSIFICATION_APIS } from './api/gs-classification-rest-api/api';
import { GS_CORE_APIS } from './api/gs-core-rest-api/api';
import { SEARCH_APIS } from './api/search-rest-api/api';
import { AUTH_APIS } from './api/auth-rest-api/api';
import { ContentApi } from './api/content-custom-api/api/content.api';
import { AlfrescoUpload } from './api-legacy/alfrescoUpload';
export class AlfrescoApiCompatibility extends AlfrescoApi {
    constructor(config) {
        super(config);
        this.core = {};
        this.activiti = {};
        this.auth = {};
        this.search = {};
        this.gsCore = {};
        this.gsClassification = {};
        this.discovery = {};
        this.initObjects();
    }
    configureJsApi(config) {
        this.setConfig(config);
    }
    initObjects() {
        this._instantiateOldObjects(CONTENT_LEGACY_APIS, this.core);
        this._instantiateOldObjects(APS_LEGACY_APIS, this.activiti);
        this._instantiateNewObjects(AUTH_APIS, this.auth);
        this._instantiateNewObjects(SEARCH_APIS, this.search);
        this._instantiateNewObjects(DISCOVERY_APIS, this.discovery);
        this._instantiateNewObjects(GS_CORE_APIS, this.gsCore);
        this._instantiateNewObjects(GS_CLASSIFICATION_APIS, this.gsClassification);
        this.nodes = this.node = this.core.nodesApi;
        this.content = new ContentApi(this);
        this.upload = new AlfrescoUpload(this);
        this.webScript = this.core.webscriptApi;
        this.bpmClient = this.processClient;
        this.ecmClient = this.contentClient;
    }
    _instantiateOldObjects(module, moduleCopy) {
        let classArray = Object.keys(module);
        classArray.forEach((currentClass) => {
            moduleCopy[currentClass] = module[currentClass];
            let obj = this._stringToObject(currentClass, module);
            let nameObj = this._lowerFirst(currentClass);
            obj === null || obj === void 0 ? void 0 : obj.init(this);
            moduleCopy[nameObj] = obj;
        });
    }
    _stringToObject(nameClass, module) {
        try {
            if (typeof module[nameClass] === 'function') {
                return new module[nameClass]();
            }
        }
        catch (error) {
            console.log(nameClass + '  ' + error);
        }
    }
    _instantiateNewObjects(module, moduleCopy) {
        let classArray = Object.keys(module);
        classArray.forEach((currentClass) => {
            moduleCopy[currentClass] = module[currentClass];
            let nameObj = this._lowerFirst(currentClass);
            moduleCopy[nameObj] = new module[currentClass](this);
        });
    }
    _lowerFirst(text) {
        return text.charAt(0).toLowerCase() + text.slice(1);
    }
}
