// @alfresco/js-api/src/api-legacy/activiti-rest-api/src/index.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.Activiti = exports.APS_LEGACY_APIS = void 0;
const AboutApi_1 = require("./api/AboutApi");
const AlfrescoApiActiviti_1 = require("./api/AlfrescoApiActiviti");
const AdminEndpointsApi_1 = require("./api/AdminEndpointsApi");
const AdminGroupsApi_1 = require("./api/AdminGroupsApi");
const AdminTenantsApi_1 = require("./api/AdminTenantsApi");
const AdminUsersApi_1 = require("./api/AdminUsersApi");
const AppsApi_1 = require("./api/AppsApi");
const AppsDefinitionApi_1 = require("./api/AppsDefinitionApi");
const AppsRuntimeApi_1 = require("./api/AppsRuntimeApi");
const CommentsApi_1 = require("./api/CommentsApi");
const ContentApi_1 = require("./api/ContentApi");
const ContentRenditionApi_1 = require("./api/ContentRenditionApi");
const EditorApi_1 = require("./api/EditorApi");
const GroupsApi_1 = require("./api/GroupsApi");
const IDMSyncApi_1 = require("./api/IDMSyncApi");
const IntegrationAccountApi_1 = require("./api/IntegrationAccountApi");
const IntegrationAlfrescoCloudApi_1 = require("./api/IntegrationAlfrescoCloudApi");
const IntegrationAlfrescoOnPremiseApi_1 = require("./api/IntegrationAlfrescoOnPremiseApi");
const IntegrationBoxApi_1 = require("./api/IntegrationBoxApi");
const IntegrationDriveApi_1 = require("./api/IntegrationDriveApi");
const ModelBpmnApi_1 = require("./api/ModelBpmnApi");
const ModelJsonBpmnApi_1 = require("./api/ModelJsonBpmnApi");
const ModelsApi_1 = require("./api/ModelsApi");
const ModelsHistoryApi_1 = require("./api/ModelsHistoryApi");
const ProcessApi_1 = require("./api/ProcessApi");
const ProcessDefinitionsApi_1 = require("./api/ProcessDefinitionsApi");
const ProcessDefinitionsFormApi_1 = require("./api/ProcessDefinitionsFormApi");
const ProcessInstancesApi_1 = require("./api/ProcessInstancesApi");
const ProcessInstancesInformationApi_1 = require("./api/ProcessInstancesInformationApi");
const ProcessInstancesListingApi_1 = require("./api/ProcessInstancesListingApi");
const ProcessInstanceVariablesApi_1 = require("./api/ProcessInstanceVariablesApi");
const ProcessScopeApi_1 = require("./api/ProcessScopeApi");
const ProfileApi_1 = require("./api/ProfileApi");
const ScriptFileApi_1 = require("./api/ScriptFileApi");
const SystemPropertiesApi_1 = require("./api/SystemPropertiesApi");
const TaskApi_1 = require("./api/TaskApi");
const TaskActionsApi_1 = require("./api/TaskActionsApi");
const TaskCheckListApi_1 = require("./api/TaskCheckListApi");
const TaskFormsApi_1 = require("./api/TaskFormsApi");
const TemporaryApi_1 = require("./api/TemporaryApi");
const UserApi_1 = require("./api/UserApi");
const UserFiltersApi_1 = require("./api/UserFiltersApi");
const UsersWorkflowApi_1 = require("./api/UsersWorkflowApi");
const ReportApi_1 = require("./api/ReportApi");
const AlfrescoApiActiviti_2 = require("./api/AlfrescoApiActiviti");
const AboutApi_2 = require("./api/AboutApi");
const AdminEndpointsApi_2 = require("./api/AdminEndpointsApi");
const AdminGroupsApi_2 = require("./api/AdminGroupsApi");
const AdminTenantsApi_2 = require("./api/AdminTenantsApi");
const AdminUsersApi_2 = require("./api/AdminUsersApi");
const AppsApi_2 = require("./api/AppsApi");
const AppsDefinitionApi_2 = require("./api/AppsDefinitionApi");
const AppsRuntimeApi_2 = require("./api/AppsRuntimeApi");
const CommentsApi_2 = require("./api/CommentsApi");
const ContentApi_2 = require("./api/ContentApi");
const ContentRenditionApi_2 = require("./api/ContentRenditionApi");
const EditorApi_2 = require("./api/EditorApi");
const GroupsApi_2 = require("./api/GroupsApi");
const IDMSyncApi_2 = require("./api/IDMSyncApi");
const IntegrationAccountApi_2 = require("./api/IntegrationAccountApi");
const IntegrationAlfrescoCloudApi_2 = require("./api/IntegrationAlfrescoCloudApi");
const IntegrationAlfrescoOnPremiseApi_2 = require("./api/IntegrationAlfrescoOnPremiseApi");
const IntegrationBoxApi_2 = require("./api/IntegrationBoxApi");
const IntegrationDriveApi_2 = require("./api/IntegrationDriveApi");
const ModelBpmnApi_2 = require("./api/ModelBpmnApi");
const ModelJsonBpmnApi_2 = require("./api/ModelJsonBpmnApi");
const ModelsApi_2 = require("./api/ModelsApi");
const ModelsHistoryApi_2 = require("./api/ModelsHistoryApi");
const ProcessApi_2 = require("./api/ProcessApi");
const ProcessDefinitionsApi_2 = require("./api/ProcessDefinitionsApi");
const ProcessDefinitionsFormApi_2 = require("./api/ProcessDefinitionsFormApi");
const ProcessInstancesApi_2 = require("./api/ProcessInstancesApi");
const ProcessInstancesInformationApi_2 = require("./api/ProcessInstancesInformationApi");
const ProcessInstancesListingApi_2 = require("./api/ProcessInstancesListingApi");
const ProcessInstanceVariablesApi_2 = require("./api/ProcessInstanceVariablesApi");
const ProcessScopeApi_2 = require("./api/ProcessScopeApi");
const ProfileApi_2 = require("./api/ProfileApi");
const ScriptFileApi_2 = require("./api/ScriptFileApi");
const SystemPropertiesApi_2 = require("./api/SystemPropertiesApi");
const TaskApi_2 = require("./api/TaskApi");
const TaskActionsApi_2 = require("./api/TaskActionsApi");
const TaskCheckListApi_2 = require("./api/TaskCheckListApi");
const TaskFormsApi_2 = require("./api/TaskFormsApi");
const TemporaryApi_2 = require("./api/TemporaryApi");
const UserApi_2 = require("./api/UserApi");
const UserFiltersApi_2 = require("./api/UserFiltersApi");
const UsersWorkflowApi_2 = require("./api/UsersWorkflowApi");
const ReportApi_2 = require("./api/ReportApi");
exports.APS_LEGACY_APIS = {
    AlfrescoApi: AlfrescoApiActiviti_1.AlfrescoApiActiviti,
    AboutApi: AboutApi_1.AboutApi,
    AdminEndpointsApi: AdminEndpointsApi_1.AdminEndpointsApi,
    AdminGroupsApi: AdminGroupsApi_1.AdminGroupsApi,
    AdminTenantsApi: AdminTenantsApi_1.AdminTenantsApi,
    AdminUsersApi: AdminUsersApi_1.AdminUsersApi,
    AppsApi: AppsApi_1.AppsApi,
    AppsDefinitionApi: AppsDefinitionApi_1.AppsDefinitionApi,
    AppsRuntimeApi: AppsRuntimeApi_1.AppsRuntimeApi,
    CommentsApi: CommentsApi_1.CommentsApi,
    ContentApi: ContentApi_1.ContentApi,
    ContentRenditionApi: ContentRenditionApi_1.ContentRenditionApi,
    EditorApi: EditorApi_1.EditorApi,
    GroupsApi: GroupsApi_1.GroupsApi,
    IDMSyncApi: IDMSyncApi_1.IDMSyncApi,
    IntegrationAccountApi: IntegrationAccountApi_1.IntegrationAccountApi,
    IntegrationAlfrescoCloudApi: IntegrationAlfrescoCloudApi_1.IntegrationAlfrescoCloudApi,
    IntegrationAlfrescoOnPremiseApi: IntegrationAlfrescoOnPremiseApi_1.IntegrationAlfrescoOnPremiseApi,
    IntegrationBoxApi: IntegrationBoxApi_1.IntegrationBoxApi,
    IntegrationDriveApi: IntegrationDriveApi_1.IntegrationDriveApi,
    ModelBpmnApi: ModelBpmnApi_1.ModelBpmnApi,
    ModelJsonBpmnApi: ModelJsonBpmnApi_1.ModelJsonBpmnApi,
    ModelsApi: ModelsApi_1.ModelsApi,
    ModelsHistoryApi: ModelsHistoryApi_1.ModelsHistoryApi,
    ProcessApi: ProcessApi_1.ProcessApi,
    ProcessDefinitionsApi: ProcessDefinitionsApi_1.ProcessDefinitionsApi,
    ProcessDefinitionsFormApi: ProcessDefinitionsFormApi_1.ProcessDefinitionsFormApi,
    ProcessInstancesApi: ProcessInstancesApi_1.ProcessInstancesApi,
    ProcessInstancesInformationApi: ProcessInstancesInformationApi_1.ProcessInstancesInformationApi,
    ProcessInstancesListingApi: ProcessInstancesListingApi_1.ProcessInstancesListingApi,
    ProcessInstanceVariablesApi: ProcessInstanceVariablesApi_1.ProcessInstanceVariablesApi,
    ProcessScopeApi: ProcessScopeApi_1.ProcessScopeApi,
    ProfileApi: ProfileApi_1.ProfileApi,
    ScriptFileApi: ScriptFileApi_1.ScriptFileApi,
    SystemPropertiesApi: SystemPropertiesApi_1.SystemPropertiesApi,
    TaskApi: TaskApi_1.TaskApi,
    TaskActionsApi: TaskActionsApi_1.TaskActionsApi,
    TaskCheckListApi: TaskCheckListApi_1.TaskCheckListApi,
    TaskFormsApi: TaskFormsApi_1.TaskFormsApi,
    TemporaryApi: TemporaryApi_1.TemporaryApi,
    UserApi: UserApi_1.UserApi,
    UserFiltersApi: UserFiltersApi_1.UserFiltersApi,
    UsersWorkflowApi: UsersWorkflowApi_1.UsersWorkflowApi,
    ReportApi: ReportApi_1.ReportApi
};
var Activiti;
(function (Activiti) {
    class AlfrescoApi extends AlfrescoApiActiviti_2.AlfrescoApiActiviti {
    }
    Activiti.AlfrescoApi = AlfrescoApi;
    class AboutApi extends AboutApi_2.AboutApi {
    }
    Activiti.AboutApi = AboutApi;
    class AdminEndpointsApi extends AdminEndpointsApi_2.AdminEndpointsApi {
    }
    Activiti.AdminEndpointsApi = AdminEndpointsApi;
    class AdminGroupsApi extends AdminGroupsApi_2.AdminGroupsApi {
    }
    Activiti.AdminGroupsApi = AdminGroupsApi;
    class AdminTenantsApi extends AdminTenantsApi_2.AdminTenantsApi {
    }
    Activiti.AdminTenantsApi = AdminTenantsApi;
    class AdminUsersApi extends AdminUsersApi_2.AdminUsersApi {
    }
    Activiti.AdminUsersApi = AdminUsersApi;
    class AppsApi extends AppsApi_2.AppsApi {
    }
    Activiti.AppsApi = AppsApi;
    class AppsDefinitionApi extends AppsDefinitionApi_2.AppsDefinitionApi {
    }
    Activiti.AppsDefinitionApi = AppsDefinitionApi;
    class AppsRuntimeApi extends AppsRuntimeApi_2.AppsRuntimeApi {
    }
    Activiti.AppsRuntimeApi = AppsRuntimeApi;
    class CommentsApi extends CommentsApi_2.CommentsApi {
    }
    Activiti.CommentsApi = CommentsApi;
    class ContentApi extends ContentApi_2.ContentApi {
    }
    Activiti.ContentApi = ContentApi;
    class ContentRenditionApi extends ContentRenditionApi_2.ContentRenditionApi {
    }
    Activiti.ContentRenditionApi = ContentRenditionApi;
    class EditorApi extends EditorApi_2.EditorApi {
    }
    Activiti.EditorApi = EditorApi;
    class GroupsApi extends GroupsApi_2.GroupsApi {
    }
    Activiti.GroupsApi = GroupsApi;
    class IDMSyncApi extends IDMSyncApi_2.IDMSyncApi {
    }
    Activiti.IDMSyncApi = IDMSyncApi;
    class IntegrationAccountApi extends IntegrationAccountApi_2.IntegrationAccountApi {
    }
    Activiti.IntegrationAccountApi = IntegrationAccountApi;
    class IntegrationAlfrescoCloudApi extends IntegrationAlfrescoCloudApi_2.IntegrationAlfrescoCloudApi {
    }
    Activiti.IntegrationAlfrescoCloudApi = IntegrationAlfrescoCloudApi;
    class IntegrationAlfrescoOnPremiseApi extends IntegrationAlfrescoOnPremiseApi_2.IntegrationAlfrescoOnPremiseApi {
    }
    Activiti.IntegrationAlfrescoOnPremiseApi = IntegrationAlfrescoOnPremiseApi;
    class IntegrationBoxApi extends IntegrationBoxApi_2.IntegrationBoxApi {
    }
    Activiti.IntegrationBoxApi = IntegrationBoxApi;
    class IntegrationDriveApi extends IntegrationDriveApi_2.IntegrationDriveApi {
    }
    Activiti.IntegrationDriveApi = IntegrationDriveApi;
    class ModelBpmnApi extends ModelBpmnApi_2.ModelBpmnApi {
    }
    Activiti.ModelBpmnApi = ModelBpmnApi;
    class ModelJsonBpmnApi extends ModelJsonBpmnApi_2.ModelJsonBpmnApi {
    }
    Activiti.ModelJsonBpmnApi = ModelJsonBpmnApi;
    class ModelsApi extends ModelsApi_2.ModelsApi {
    }
    Activiti.ModelsApi = ModelsApi;
    class ModelsHistoryApi extends ModelsHistoryApi_2.ModelsHistoryApi {
    }
    Activiti.ModelsHistoryApi = ModelsHistoryApi;
    class ProcessApi extends ProcessApi_2.ProcessApi {
    }
    Activiti.ProcessApi = ProcessApi;
    class ProcessDefinitionsApi extends ProcessDefinitionsApi_2.ProcessDefinitionsApi {
    }
    Activiti.ProcessDefinitionsApi = ProcessDefinitionsApi;
    class ProcessDefinitionsFormApi extends ProcessDefinitionsFormApi_2.ProcessDefinitionsFormApi {
    }
    Activiti.ProcessDefinitionsFormApi = ProcessDefinitionsFormApi;
    class ProcessInstancesApi extends ProcessInstancesApi_2.ProcessInstancesApi {
    }
    Activiti.ProcessInstancesApi = ProcessInstancesApi;
    class ProcessInstancesInformationApi extends ProcessInstancesInformationApi_2.ProcessInstancesInformationApi {
    }
    Activiti.ProcessInstancesInformationApi = ProcessInstancesInformationApi;
    class ProcessInstancesListingApi extends ProcessInstancesListingApi_2.ProcessInstancesListingApi {
    }
    Activiti.ProcessInstancesListingApi = ProcessInstancesListingApi;
    class ProcessInstanceVariablesApi extends ProcessInstanceVariablesApi_2.ProcessInstanceVariablesApi {
    }
    Activiti.ProcessInstanceVariablesApi = ProcessInstanceVariablesApi;
    class ProcessScopeApi extends ProcessScopeApi_2.ProcessScopeApi {
    }
    Activiti.ProcessScopeApi = ProcessScopeApi;
    class ProfileApi extends ProfileApi_2.ProfileApi {
    }
    Activiti.ProfileApi = ProfileApi;
    class ScriptFileApi extends ScriptFileApi_2.ScriptFileApi {
    }
    Activiti.ScriptFileApi = ScriptFileApi;
    class SystemPropertiesApi extends SystemPropertiesApi_2.SystemPropertiesApi {
    }
    Activiti.SystemPropertiesApi = SystemPropertiesApi;
    class TaskApi extends TaskApi_2.TaskApi {
    }
    Activiti.TaskApi = TaskApi;
    class TaskActionsApi extends TaskActionsApi_2.TaskActionsApi {
    }
    Activiti.TaskActionsApi = TaskActionsApi;
    class TaskCheckListApi extends TaskCheckListApi_2.TaskCheckListApi {
    }
    Activiti.TaskCheckListApi = TaskCheckListApi;
    class TaskFormsApi extends TaskFormsApi_2.TaskFormsApi {
    }
    Activiti.TaskFormsApi = TaskFormsApi;
    class TemporaryApi extends TemporaryApi_2.TemporaryApi {
    }
    Activiti.TemporaryApi = TemporaryApi;
    class UserApi extends UserApi_2.UserApi {
    }
    Activiti.UserApi = UserApi;
    class UserFiltersApi extends UserFiltersApi_2.UserFiltersApi {
    }
    Activiti.UserFiltersApi = UserFiltersApi;
    class UsersWorkflowApi extends UsersWorkflowApi_2.UsersWorkflowApi {
    }
    Activiti.UsersWorkflowApi = UsersWorkflowApi;
    class ReportApi extends ReportApi_2.ReportApi {
    }
    Activiti.ReportApi = ReportApi;
})(Activiti = exports.Activiti || (exports.Activiti = {}));
