// @alfresco/js-api/src/api-legacy/content-rest-api/src/api/childAssociationsApi.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChildAssociationsApi = void 0;
const nodes_api_1 = require("../../../../api/content-rest-api/api/nodes.api");
class ChildAssociationsApi {
    init(alfrescoApi) {
        this.nodesApi = new nodes_api_1.NodesApi(alfrescoApi);
    }
    addNode(nodeId, nodeBody, opts) {
        return this.nodesApi.createNode(nodeId, nodeBody, opts);
    }
    addSecondaryChildAssoc(parentId, assocChildBody) {
        return this.nodesApi.createSecondaryChildAssociation(parentId, assocChildBody);
    }
    deleteNode(nodeId, opts) {
        return this.nodesApi.deleteNode(nodeId, opts);
    }
    getNodeChildren(nodeId, opts) {
        return this.nodesApi.listNodeChildren(nodeId, opts);
    }
    listParents(childId, opts) {
        return this.nodesApi.listParents(childId, opts);
    }
    listSecondaryChildAssociations(parentId, opts) {
        return this.nodesApi.listSecondaryChildren(parentId, opts);
    }
    moveNode(nodeId, moveBody, opts) {
        return this.nodesApi.moveNode(nodeId, moveBody, opts);
    }
    removeSecondaryChildAssoc(parentId, childId, opts) {
        return this.nodesApi.deleteSecondaryChildAssociation(parentId, childId, opts);
    }
}
exports.ChildAssociationsApi = ChildAssociationsApi;
