// @alfresco/js-api/src/api-legacy/content-rest-api/src/api/commentsApi.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommentsApi = void 0;
const comments_api_1 = require("../../../../api/content-rest-api/api/comments.api");
class CommentsApi {
    init(alfrescoApi) {
        this.commentsApi = new comments_api_1.CommentsApi(alfrescoApi);
    }
    addComment(nodeId, commentBody, opts) {
        return this.commentsApi.createComment(nodeId, commentBody, opts);
    }
    getComments(nodeId, opts) {
        return this.commentsApi.listComments(nodeId, opts);
    }
    removeComment(nodeId, commentId) {
        return this.commentsApi.deleteComment(nodeId, commentId);
    }
    updateComment(nodeId, commentId, commentBody, opts) {
        return this.commentsApi.updateComment(nodeId, commentId, commentBody, opts);
    }
}
exports.CommentsApi = CommentsApi;
