// @alfresco/js-api/src/api-legacy/content-rest-api/src/api/networksApi.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworksApi = void 0;
const networks_api_1 = require("../../../../api/content-rest-api/api/networks.api");
class NetworksApi {
    init(alfrescoApi) {
        this.networksApi = new networks_api_1.NetworksApi(alfrescoApi);
    }
    getNetwork(networkId, opts) {
        return this.networksApi.getNetwork(networkId, opts);
    }
    getNetworkForPerson(personId, networkId, opts) {
        return this.networksApi.getNetworkForPerson(personId, networkId, opts);
    }
    listNetworksForPerson(personId, opts) {
        return this.networksApi.listNetworksForPerson(personId, opts);
    }
}
exports.NetworksApi = NetworksApi;
