// @alfresco/js-api/src/api-legacy/content-rest-api/src/api/ratingsApi.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.RatingsApi = void 0;
const ratings_api_1 = require("../../../../api/content-rest-api/api/ratings.api");
class RatingsApi {
    init(alfrescoApi) {
        this.ratingsApi = new ratings_api_1.RatingsApi(alfrescoApi);
    }
    getRating(nodeId, ratingId, opts) {
        return this.ratingsApi.getRating(nodeId, ratingId, opts);
    }
    getRatings(nodeId, opts) {
        return this.ratingsApi.listRatings(nodeId, opts);
    }
    rate(nodeId, ratingBody) {
        return this.ratingsApi.createRating(nodeId, ratingBody);
    }
    removeRating(nodeId, ratingId) {
        return this.ratingsApi.deleteRating(nodeId, ratingId);
    }
}
exports.RatingsApi = RatingsApi;
