// @alfresco/js-api/src/api-legacy/content-rest-api/src/api/tagsApi.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagsApi = void 0;
const tags_api_1 = require("../../../../api/content-rest-api/api/tags.api");
class TagsApi {
    init(alfrescoApi) {
        this.tagsApi = new tags_api_1.TagsApi(alfrescoApi);
    }
    addTag(nodeId, tagBody) {
        return this.tagsApi.createTagForNode(nodeId, tagBody);
    }
    getNodeTags(nodeId, opts) {
        return this.tagsApi.listTagsForNode(nodeId, opts);
    }
    getTag(tagId, opts) {
        return this.tagsApi.getTag(tagId, opts);
    }
    getTags(opts) {
        return this.tagsApi.listTags(opts);
    }
    removeTag(nodeId, tagId) {
        return this.tagsApi.deleteTagFromNode(nodeId, tagId);
    }
    updateTag(tagId, tagBody) {
        return this.tagsApi.updateTag(tagId, tagBody);
    }
}
exports.TagsApi = TagsApi;
