// @alfresco/js-api/src/api/activiti-rest-api/model/historicProcessInstanceQueryRepresentation.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.HistoricProcessInstanceQueryRepresentation = void 0;
const dateAlfresco_1 = require("../../content-custom-api/model/dateAlfresco");
const queryVariable_1 = require("./queryVariable");
class HistoricProcessInstanceQueryRepresentation {
    constructor(input) {
        if (input) {
            Object.assign(this, input);
            this.finishedAfter = input.finishedAfter ? dateAlfresco_1.DateAlfresco.parseDate(input.finishedAfter) : undefined;
            this.finishedBefore = input.finishedBefore ? dateAlfresco_1.DateAlfresco.parseDate(input.finishedBefore) : undefined;
            this.startedAfter = input.startedAfter ? dateAlfresco_1.DateAlfresco.parseDate(input.startedAfter) : undefined;
            this.startedBefore = input.startedBefore ? dateAlfresco_1.DateAlfresco.parseDate(input.startedBefore) : undefined;
            if (input.variables) {
                this.variables = input.variables.map((item) => {
                    return new queryVariable_1.QueryVariable(item);
                });
            }
        }
    }
}
exports.HistoricProcessInstanceQueryRepresentation = HistoricProcessInstanceQueryRepresentation;
