// @alfresco/js-api/src/api/activiti-rest-api/model/historicTaskInstanceQueryRepresentation.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.HistoricTaskInstanceQueryRepresentation = void 0;
const dateAlfresco_1 = require("../../content-custom-api/model/dateAlfresco");
const queryVariable_1 = require("./queryVariable");
class HistoricTaskInstanceQueryRepresentation {
    constructor(input) {
        if (input) {
            Object.assign(this, input);
            this.dueDate = input.dueDate ? dateAlfresco_1.DateAlfresco.parseDate(input.dueDate) : undefined;
            this.dueDateAfter = input.dueDateAfter ? dateAlfresco_1.DateAlfresco.parseDate(input.dueDateAfter) : undefined;
            this.dueDateBefore = input.dueDateBefore ? dateAlfresco_1.DateAlfresco.parseDate(input.dueDateBefore) : undefined;
            if (input.processVariables) {
                this.processVariables = input.processVariables.map((item) => {
                    return new queryVariable_1.QueryVariable(item);
                });
            }
            this.taskCompletedAfter = input.taskCompletedAfter ? dateAlfresco_1.DateAlfresco.parseDate(input.taskCompletedAfter) : undefined;
            this.taskCompletedBefore = input.taskCompletedBefore ? dateAlfresco_1.DateAlfresco.parseDate(input.taskCompletedBefore) : undefined;
            this.taskCompletedOn = input.taskCompletedOn ? dateAlfresco_1.DateAlfresco.parseDate(input.taskCompletedOn) : undefined;
            this.taskCreatedAfter = input.taskCreatedAfter ? dateAlfresco_1.DateAlfresco.parseDate(input.taskCreatedAfter) : undefined;
            this.taskCreatedBefore = input.taskCreatedBefore ? dateAlfresco_1.DateAlfresco.parseDate(input.taskCreatedBefore) : undefined;
            this.taskCreatedOn = input.taskCreatedOn ? dateAlfresco_1.DateAlfresco.parseDate(input.taskCreatedOn) : undefined;
            if (input.taskVariables) {
                this.taskVariables = input.taskVariables.map((item) => {
                    return new queryVariable_1.QueryVariable(item);
                });
            }
        }
    }
}
exports.HistoricTaskInstanceQueryRepresentation = HistoricTaskInstanceQueryRepresentation;
