// @alfresco/js-api/src/api/activiti-rest-api/model/taskFilterRepresentation.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskFilterRepresentation = void 0;
const dateAlfresco_1 = require("../../content-custom-api/model/dateAlfresco");
class TaskFilterRepresentation {
    constructor(input) {
        if (input) {
            Object.assign(this, input);
            this.dueAfter = input.dueAfter ? dateAlfresco_1.DateAlfresco.parseDate(input.dueAfter) : undefined;
            this.dueBefore = input.dueBefore ? dateAlfresco_1.DateAlfresco.parseDate(input.dueBefore) : undefined;
        }
    }
}
exports.TaskFilterRepresentation = TaskFilterRepresentation;
(function (TaskFilterRepresentation) {
    TaskFilterRepresentation.StateEnum = {
        Active: 'active',
        Completed: 'completed',
        All: 'all'
    };
})(TaskFilterRepresentation = exports.TaskFilterRepresentation || (exports.TaskFilterRepresentation = {}));
