// @alfresco/js-api/src/api/activiti-rest-api/model/userRepresentation.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserRepresentation = void 0;
const dateAlfresco_1 = require("../../content-custom-api/model/dateAlfresco");
const groupRepresentation_1 = require("./groupRepresentation");
const lightAppRepresentation_1 = require("./lightAppRepresentation");
class UserRepresentation {
    constructor(input) {
        if (input) {
            Object.assign(this, input);
            if (input.apps) {
                this.apps = input.apps.map((item) => {
                    return new lightAppRepresentation_1.LightAppRepresentation(item);
                });
            }
            this.created = input.created ? dateAlfresco_1.DateAlfresco.parseDate(input.created) : undefined;
            if (input.groups) {
                this.groups = input.groups.map((item) => {
                    return new groupRepresentation_1.GroupRepresentation(item);
                });
            }
            this.lastUpdate = input.lastUpdate ? dateAlfresco_1.DateAlfresco.parseDate(input.lastUpdate) : undefined;
            this.latestSyncTimeStamp = input.latestSyncTimeStamp ? dateAlfresco_1.DateAlfresco.parseDate(input.latestSyncTimeStamp) : undefined;
            this.primaryGroup = input.primaryGroup ? new groupRepresentation_1.GroupRepresentation(input.primaryGroup) : undefined;
        }
    }
}
exports.UserRepresentation = UserRepresentation;
