// @alfresco/js-api/esm5/src/api/content-rest-api/api/renditions.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { DirectAccessUrlEntry } from '../model/directAccessUrlEntry';
import { RenditionEntry } from '../model/renditionEntry';
import { RenditionPaging } from '../model/renditionPaging';
import { BaseApi } from './base.api';
import { throwIfNotDefined } from '../../../assert';
var RenditionsApi = (function (_super) {
    __extends(RenditionsApi, _super);
    function RenditionsApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RenditionsApi.prototype.createRendition = function (nodeId, renditionBodyCreate) {
        throwIfNotDefined(nodeId, 'nodeId');
        throwIfNotDefined(renditionBodyCreate, 'renditionBodyCreate');
        var postBody = renditionBodyCreate;
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/renditions', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    RenditionsApi.prototype.getRendition = function (nodeId, renditionId) {
        throwIfNotDefined(nodeId, 'nodeId');
        throwIfNotDefined(renditionId, 'renditionId');
        var postBody = null;
        var pathParams = {
            'nodeId': nodeId, 'renditionId': renditionId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/renditions/{renditionId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RenditionEntry);
    };
    RenditionsApi.prototype.getRenditionContent = function (nodeId, renditionId, opts) {
        throwIfNotDefined(nodeId, 'nodeId');
        throwIfNotDefined(renditionId, 'renditionId');
        opts = opts || {};
        var postBody = null;
        var pathParams = {
            'nodeId': nodeId, 'renditionId': renditionId
        };
        var queryParams = {
            'attachment': opts['attachment'],
            'placeholder': opts['placeholder']
        };
        var headerParams = {
            'If-Modified-Since': opts['ifModifiedSince'], 'Range': opts['range']
        };
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/octet-stream'];
        return this.apiClient.callApi('/nodes/{nodeId}/renditions/{renditionId}/content', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, 'blob');
    };
    RenditionsApi.prototype.listRenditions = function (nodeId, opts) {
        throwIfNotDefined(nodeId, 'nodeId');
        opts = opts || {};
        var postBody = null;
        var pathParams = {
            'nodeId': nodeId
        };
        var queryParams = {
            'where': opts['where']
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/renditions', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, RenditionPaging);
    };
    RenditionsApi.prototype.requestDirectAccessUrl = function (nodeId, renditionId) {
        throwIfNotDefined(nodeId, 'nodeId');
        throwIfNotDefined(renditionId, 'renditionId');
        var pathParams = {
            'nodeId': nodeId,
            'renditionId': renditionId
        };
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/nodes/{nodeId}/renditions/{renditionId}/request-direct-access-url', 'POST', pathParams, {}, {}, {}, null, contentTypes, accepts, DirectAccessUrlEntry);
    };
    return RenditionsApi;
}(BaseApi));
export { RenditionsApi };
