// @alfresco/js-api/src/api/gs-core-rest-api/api/index.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.GS_CORE_APIS = void 0;
const tslib_1 = require("tslib");
(0, tslib_1.__exportStar)(require("./filePlans.api"), exports);
const filePlans_api_1 = require("./filePlans.api");
(0, tslib_1.__exportStar)(require("./files.api"), exports);
const files_api_1 = require("./files.api");
(0, tslib_1.__exportStar)(require("./gsSites.api"), exports);
const gsSites_api_1 = require("./gsSites.api");
(0, tslib_1.__exportStar)(require("./recordCategories.api"), exports);
const recordCategories_api_1 = require("./recordCategories.api");
(0, tslib_1.__exportStar)(require("./recordFolders.api"), exports);
const recordFolders_api_1 = require("./recordFolders.api");
(0, tslib_1.__exportStar)(require("./records.api"), exports);
const records_api_1 = require("./records.api");
(0, tslib_1.__exportStar)(require("./transferContainers.api"), exports);
const transferContainers_api_1 = require("./transferContainers.api");
(0, tslib_1.__exportStar)(require("./transfers.api"), exports);
const transfers_api_1 = require("./transfers.api");
(0, tslib_1.__exportStar)(require("./unfiledContainers.api"), exports);
const unfiledContainers_api_1 = require("./unfiledContainers.api");
(0, tslib_1.__exportStar)(require("./unfiledRecordFolders.api"), exports);
const unfiledRecordFolders_api_1 = require("./unfiledRecordFolders.api");
exports.GS_CORE_APIS = {
    FilePlansApi: filePlans_api_1.FilePlansApi,
    FilesApi: files_api_1.FilesApi,
    GsSitesApi: gsSites_api_1.GsSitesApi,
    RecordCategoriesApi: recordCategories_api_1.RecordCategoriesApi,
    RecordFoldersApi: recordFolders_api_1.RecordFoldersApi,
    RecordsApi: records_api_1.RecordsApi,
    TransferContainersApi: transferContainers_api_1.TransferContainersApi,
    TransfersApi: transfers_api_1.TransfersApi,
    UnfiledContainersApi: unfiledContainers_api_1.UnfiledContainersApi,
    UnfiledRecordFoldersApi: unfiledRecordFolders_api_1.UnfiledRecordFoldersApi
};
