// @alfresco/js-api/src/api/gs-core-rest-api/api/recordCategories.api.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecordCategoriesApi = void 0;
const recordCategoryChildEntry_1 = require("../model/recordCategoryChildEntry");
const recordCategoryChildPaging_1 = require("../model/recordCategoryChildPaging");
const recordCategoryEntry_1 = require("../model/recordCategoryEntry");
const base_api_1 = require("./base.api");
const alfrescoApiClient_1 = require("../../../alfrescoApiClient");
const assert_1 = require("../../../assert");
class RecordCategoriesApi extends base_api_1.BaseApi {
    createRecordCategoryChild(recordCategoryId, nodeBodyCreate, opts) {
        (0, assert_1.throwIfNotDefined)(recordCategoryId, 'recordCategoryId');
        (0, assert_1.throwIfNotDefined)(nodeBodyCreate, 'nodeBodyCreate');
        opts = opts || {};
        let postBody = nodeBodyCreate;
        let pathParams = {
            'recordCategoryId': recordCategoryId
        };
        let queryParams = {
            'autoRename': opts['autoRename'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json', 'multipart/form-data'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}/children', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, recordCategoryChildEntry_1.RecordCategoryChildEntry);
    }
    deleteRecordCategory(recordCategoryId) {
        (0, assert_1.throwIfNotDefined)(recordCategoryId, 'recordCategoryId');
        let postBody = null;
        let pathParams = {
            'recordCategoryId': recordCategoryId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    getRecordCategory(recordCategoryId, opts) {
        (0, assert_1.throwIfNotDefined)(recordCategoryId, 'recordCategoryId');
        opts = opts || {};
        let postBody = null;
        let pathParams = {
            'recordCategoryId': recordCategoryId
        };
        let queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, recordCategoryEntry_1.RecordCategoryEntry);
    }
    listRecordCategoryChildren(recordCategoryId, opts) {
        (0, assert_1.throwIfNotDefined)(recordCategoryId, 'recordCategoryId');
        opts = opts || {};
        let postBody = null;
        let pathParams = {
            'recordCategoryId': recordCategoryId
        };
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'where': opts['where'],
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'includeSource': opts['includeSource'],
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}/children', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, recordCategoryChildPaging_1.RecordCategoryChildPaging);
    }
    updateRecordCategory(recordCategoryId, recordCategoryBodyUpdate, opts) {
        (0, assert_1.throwIfNotDefined)(recordCategoryId, 'recordCategoryId');
        (0, assert_1.throwIfNotDefined)(recordCategoryBodyUpdate, 'recordCategoryBodyUpdate');
        opts = opts || {};
        let postBody = recordCategoryBodyUpdate;
        let pathParams = {
            'recordCategoryId': recordCategoryId
        };
        let queryParams = {
            'include': (0, alfrescoApiClient_1.buildCollectionParam)(opts['include'], 'csv'),
            'fields': (0, alfrescoApiClient_1.buildCollectionParam)(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/record-categories/{recordCategoryId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, recordCategoryEntry_1.RecordCategoryEntry);
    }
}
exports.RecordCategoriesApi = RecordCategoriesApi;
