// @alfresco/js-api/esm5/src/contentClient.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { AlfrescoApiClient } from './alfrescoApiClient';
var ContentClient = (function (_super) {
    __extends(ContentClient, _super);
    function ContentClient(config, servicePath) {
        var _this = _super.call(this) || this;
        _this.className = 'ContentClient';
        _this.setConfig(config, servicePath);
        return _this;
    }
    ContentClient.prototype.setConfig = function (config, servicePath) {
        this.config = config;
        this.servicePath = servicePath;
        this.changeHost();
    };
    ContentClient.prototype.changeHost = function () {
        this.host = this.config.hostEcm;
        this.basePath = "".concat(this.config.hostEcm, "/").concat(this.config.contextRoot).concat(this.servicePath);
    };
    ContentClient.prototype.setAuthentications = function (authentications) {
        this.authentications = authentications;
    };
    return ContentClient;
}(AlfrescoApiClient));
export { ContentClient };
