// @alfresco/js-api/esm5/src/processClient.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { AlfrescoApiClient } from './alfrescoApiClient';
var ProcessClient = (function (_super) {
    __extends(ProcessClient, _super);
    function ProcessClient(config) {
        var _this = _super.call(this) || this;
        _this.className = 'ProcessClient';
        _this.setConfig(config);
        return _this;
    }
    ProcessClient.prototype.setConfig = function (config) {
        this.config = config;
        this.changeHost();
    };
    ProcessClient.prototype.changeHost = function () {
        this.host = this.config.hostBpm;
        this.basePath = "".concat(this.config.hostBpm, "/").concat(this.config.contextRootBpm);
    };
    ProcessClient.prototype.setAuthentications = function (authentications) {
        this.authentications = authentications;
    };
    return ProcessClient;
}(AlfrescoApiClient));
export { ProcessClient };
