// @alfresco/js-api/esm5/src/storage.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
var Storage = (function () {
    function Storage() {
        if (this.supportsStorage()) {
            this._storage = window.localStorage;
        }
    }
    Storage.prototype.supportsStorage = function () {
        try {
            return 'sessionStorage' in window && window.sessionStorage !== null;
        }
        catch (e) {
            return false;
        }
    };
    Storage.prototype.setDomainPrefix = function (prefix) {
        this.prefix = prefix ? prefix + '_' : '';
    };
    Storage.prototype.setStorage = function (storage) {
        this._storage = storage;
    };
    Storage.prototype.setItem = function (key, value) {
        if (this.supportsStorage()) {
            this._storage.setItem(this.prefix + key, value);
        }
    };
    Storage.prototype.removeItem = function (key) {
        if (this.supportsStorage()) {
            this._storage.removeItem(this.prefix + key);
        }
    };
    Storage.prototype.getItem = function (key) {
        if (this.supportsStorage()) {
            return this._storage.getItem(this.prefix + key);
        }
    };
    return Storage;
}());
export { Storage };
