// @alfresco/js-api/esm2015/src/api/gs-classification-rest-api/api/classificationReasons.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { ClassificationReasonEntry } from '../model/classificationReasonEntry';
import { ClassificationReasonsPaging } from '../model/classificationReasonsPaging';
import { BaseApi } from './base.api';
import { buildCollectionParam } from '../../../alfrescoApiClient';
import { throwIfNotDefined } from '../../../assert';
export class ClassificationReasonsApi extends BaseApi {
    createClassificationReason(classificationReason) {
        throwIfNotDefined(classificationReason, 'classificationReason');
        let postBody = classificationReason;
        let pathParams = {};
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationReasonEntry);
    }
    deleteClassificationReason(classificationReasonId) {
        throwIfNotDefined(classificationReasonId, 'classificationReasonId');
        let postBody = null;
        let pathParams = {
            'classificationReasonId': classificationReasonId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons/{classificationReasonId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    }
    listClassificationReasons(opts) {
        opts = opts || {};
        let postBody = null;
        let pathParams = {};
        let queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'fields': buildCollectionParam(opts['fields'], 'csv')
        };
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationReasonsPaging);
    }
    showClassificationReasonById(classificationReasonId) {
        throwIfNotDefined(classificationReasonId, 'classificationReasonId');
        let postBody = null;
        let pathParams = {
            'classificationReasonId': classificationReasonId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons/{classificationReasonId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationReasonEntry);
    }
    updateClassificationReason(classificationReasonId, classificationReason) {
        throwIfNotDefined(classificationReasonId, 'classificationReasonId');
        throwIfNotDefined(classificationReason, 'classificationReason');
        let postBody = classificationReason;
        let pathParams = {
            'classificationReasonId': classificationReasonId
        };
        let queryParams = {};
        let headerParams = {};
        let formParams = {};
        let contentTypes = ['application/json'];
        let accepts = ['application/json'];
        return this.apiClient.callApi('/classification-reasons/{classificationReasonId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, ClassificationReasonEntry);
    }
}
