// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/api/childAssociationsApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { NodesApi } from '../../../../api/content-rest-api/api/nodes.api';
var ChildAssociationsApi = (function () {
    function ChildAssociationsApi() {
    }
    ChildAssociationsApi.prototype.init = function (alfrescoApi) {
        this.nodesApi = new NodesApi(alfrescoApi);
    };
    ChildAssociationsApi.prototype.addNode = function (nodeId, nodeBody, opts) {
        return this.nodesApi.createNode(nodeId, nodeBody, opts);
    };
    ChildAssociationsApi.prototype.addSecondaryChildAssoc = function (parentId, assocChildBody) {
        return this.nodesApi.createSecondaryChildAssociation(parentId, assocChildBody);
    };
    ChildAssociationsApi.prototype.deleteNode = function (nodeId, opts) {
        return this.nodesApi.deleteNode(nodeId, opts);
    };
    ChildAssociationsApi.prototype.getNodeChildren = function (nodeId, opts) {
        return this.nodesApi.listNodeChildren(nodeId, opts);
    };
    ChildAssociationsApi.prototype.listParents = function (childId, opts) {
        return this.nodesApi.listParents(childId, opts);
    };
    ChildAssociationsApi.prototype.listSecondaryChildAssociations = function (parentId, opts) {
        return this.nodesApi.listSecondaryChildren(parentId, opts);
    };
    ChildAssociationsApi.prototype.moveNode = function (nodeId, moveBody, opts) {
        return this.nodesApi.moveNode(nodeId, moveBody, opts);
    };
    ChildAssociationsApi.prototype.removeSecondaryChildAssoc = function (parentId, childId, opts) {
        return this.nodesApi.deleteSecondaryChildAssociation(parentId, childId, opts);
    };
    return ChildAssociationsApi;
}());
export { ChildAssociationsApi };
