// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/api/commentsApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { CommentsApi as NewCommentsApi } from '../../../../api/content-rest-api/api/comments.api';
var CommentsApi = (function () {
    function CommentsApi() {
    }
    CommentsApi.prototype.init = function (alfrescoApi) {
        this.commentsApi = new NewCommentsApi(alfrescoApi);
    };
    CommentsApi.prototype.addComment = function (nodeId, commentBody, opts) {
        return this.commentsApi.createComment(nodeId, commentBody, opts);
    };
    CommentsApi.prototype.getComments = function (nodeId, opts) {
        return this.commentsApi.listComments(nodeId, opts);
    };
    CommentsApi.prototype.removeComment = function (nodeId, commentId) {
        return this.commentsApi.deleteComment(nodeId, commentId);
    };
    CommentsApi.prototype.updateComment = function (nodeId, commentId, commentBody, opts) {
        return this.commentsApi.updateComment(nodeId, commentId, commentBody, opts);
    };
    return CommentsApi;
}());
export { CommentsApi };
