// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/api/groupsApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { GroupsApi as NewGroupsApi } from '../../../../api/content-rest-api/api/groups.api';
var GroupsApi = (function () {
    function GroupsApi() {
    }
    GroupsApi.prototype.init = function (alfrescoApi) {
        this.groupsApi = new NewGroupsApi(alfrescoApi);
    };
    GroupsApi.prototype.createGroup = function (groupBody, opts) {
        return this.groupsApi.createGroup(groupBody, opts);
    };
    GroupsApi.prototype.getGroups = function (opts) {
        return this.groupsApi.listGroups(opts);
    };
    GroupsApi.prototype.deleteGroup = function (groupId, opts) {
        return this.groupsApi.deleteGroup(groupId, opts);
    };
    GroupsApi.prototype.getGroup = function (groupId, opts) {
        return this.groupsApi.getGroup(groupId, opts);
    };
    GroupsApi.prototype.updateGroup = function (groupId, groupBody, opts) {
        return this.groupsApi.updateGroup(groupId, groupBody, opts);
    };
    GroupsApi.prototype.getGroupMembers = function (groupId, opts) {
        return this.groupsApi.listGroupMemberships(groupId, opts);
    };
    GroupsApi.prototype.addGroupMember = function (groupId, groupMemberBody, opts) {
        return this.groupsApi.createGroupMembership(groupId, groupMemberBody, opts);
    };
    GroupsApi.prototype.deleteGroupMember = function (groupId, groupMemberId) {
        return this.groupsApi.deleteGroupMembership(groupId, groupMemberId);
    };
    return GroupsApi;
}());
export { GroupsApi };
