// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/api/networksApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { NetworksApi as NewNetworksApi } from '../../../../api/content-rest-api/api/networks.api';
var NetworksApi = (function () {
    function NetworksApi() {
    }
    NetworksApi.prototype.init = function (alfrescoApi) {
        this.networksApi = new NewNetworksApi(alfrescoApi);
    };
    NetworksApi.prototype.getNetwork = function (networkId, opts) {
        return this.networksApi.getNetwork(networkId, opts);
    };
    NetworksApi.prototype.getNetworkForPerson = function (personId, networkId, opts) {
        return this.networksApi.getNetworkForPerson(personId, networkId, opts);
    };
    NetworksApi.prototype.listNetworksForPerson = function (personId, opts) {
        return this.networksApi.listNetworksForPerson(personId, opts);
    };
    return NetworksApi;
}());
export { NetworksApi };
