// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/api/nodesApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { NodesApi as NewNodesApi } from '../../../../api/content-rest-api/api/nodes.api';
import { TrashcanApi } from '../../../../api/content-rest-api';
var NodesApi = (function () {
    function NodesApi() {
    }
    NodesApi.prototype.init = function (alfrescoApi) {
        this.nodesApi = new NewNodesApi(alfrescoApi);
        this.trashcanApi = new TrashcanApi(alfrescoApi);
    };
    NodesApi.prototype.addNode = function (nodeId, nodeBody, opts) {
        return this.nodesApi.createNode(nodeId, nodeBody, opts);
    };
    NodesApi.prototype.copyNode = function (nodeId, copyBody, opts) {
        return this.nodesApi.copyNode(nodeId, copyBody, opts);
    };
    NodesApi.prototype.deleteNode = function (nodeId, opts) {
        return this.nodesApi.deleteNode(nodeId, opts);
    };
    NodesApi.prototype.getDeletedNode = function (nodeId, opts) {
        return this.trashcanApi.getDeletedNode(nodeId, opts);
    };
    NodesApi.prototype.getDeletedNodes = function (opts) {
        return this.trashcanApi.listDeletedNodes(opts);
    };
    NodesApi.prototype.getFileContent = function (nodeId, opts) {
        return this.nodesApi.getNodeContent(nodeId, opts);
    };
    NodesApi.prototype.getNode = function (nodeId, opts) {
        return this.nodesApi.getNode(nodeId, opts);
    };
    NodesApi.prototype.getNodeContent = function (nodeId, opts) {
        return this.nodesApi.getNodeContent(nodeId, opts);
    };
    NodesApi.prototype.getNodeChildren = function (nodeId, opts) {
        return this.nodesApi.listNodeChildren(nodeId, opts);
    };
    NodesApi.prototype.getParents = function (nodeId, opts) {
        return this.nodesApi.listParents(nodeId, opts);
    };
    NodesApi.prototype.getSecondaryChildren = function (nodeId, opts) {
        return this.nodesApi.listSecondaryChildren(nodeId, opts);
    };
    NodesApi.prototype.getSourceAssociations = function (nodeId, opts) {
        return this.nodesApi.listSourceAssociations(nodeId, opts);
    };
    NodesApi.prototype.getTargetAssociations = function (nodeId, opts) {
        return this.nodesApi.listTargetAssociations(nodeId, opts);
    };
    NodesApi.prototype.lockNode = function (nodeId, nodeBodyLock, opts) {
        return this.nodesApi.lockNode(nodeId, nodeBodyLock, opts);
    };
    NodesApi.prototype.unlockNode = function (nodeId, opts) {
        return this.nodesApi.unlockNode(nodeId, opts);
    };
    NodesApi.prototype.moveNode = function (nodeId, moveBody, opts) {
        return this.nodesApi.moveNode(nodeId, moveBody, opts);
    };
    NodesApi.prototype.purgeDeletedNode = function (nodeId) {
        return this.trashcanApi.deleteDeletedNode(nodeId);
    };
    NodesApi.prototype.restoreNode = function (nodeId) {
        return this.trashcanApi.restoreDeletedNode(nodeId);
    };
    NodesApi.prototype.updateFileContent = function (nodeId, contentBody, opts) {
        return this.nodesApi.updateNodeContent(nodeId, contentBody, opts);
    };
    NodesApi.prototype.updateNodeContent = function (nodeId, contentBody, opts) {
        return this.nodesApi.updateNodeContent(nodeId, contentBody, opts);
    };
    NodesApi.prototype.updateNode = function (nodeId, nodeBody, opts) {
        return this.nodesApi.updateNode(nodeId, nodeBody, opts);
    };
    NodesApi.prototype.getNodeInfo = function (nodeId, opts) {
        var _this = this;
        return new Promise(function (resolve, reject) {
            _this.getNode(nodeId, opts).then(function (nodeEntry) {
                resolve(nodeEntry.entry);
            }, function (error) {
                reject(error);
            });
        });
    };
    NodesApi.prototype.deleteNodePermanent = function (nodeId) {
        return this.deleteNode(nodeId, { permanent: true });
    };
    NodesApi.prototype.createFolder = function (name, relativePath, nodeId, opts) {
        nodeId = nodeId || '-root-';
        var nodeBody = {
            'name': name,
            'nodeType': 'cm:folder',
            'relativePath': relativePath
        };
        return this.addNode(nodeId, nodeBody, opts);
    };
    NodesApi.prototype.createFolderAutoRename = function (name, relativePath, nodeId, opts) {
        var optAutoRename = { autoRename: true };
        opts = opts || {};
        opts = Object.assign(opts, optAutoRename);
        return this.createFolder(name, relativePath, nodeId, opts);
    };
    return NodesApi;
}());
export { NodesApi };
