// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/api/peopleApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { PeopleApi as PeopleApiNew } from '../../../../api/content-rest-api/api/people.api';
import { NetworksApi } from '../../../../api/content-rest-api/api/networks.api';
import { FavoritesApi } from '../../../../api/content-rest-api/api/favorites.api';
import { SitesApi } from '../../../../api/content-rest-api/api/sites.api';
import { PreferencesApi } from '../../../../api/content-rest-api/api/preferences.api';
import { ActivitiesApi } from '../../../../api/content-rest-api/api/activities.api';
import { GroupsApi } from '../../../../api/content-rest-api/api/groups.api';
var PeopleApi = (function () {
    function PeopleApi() {
    }
    PeopleApi.prototype.init = function (alfrescoApi) {
        this.networksApi = new NetworksApi(alfrescoApi);
        this.peopleApiNew = new PeopleApiNew(alfrescoApi);
        this.favoritesApi = new FavoritesApi(alfrescoApi);
        this.sitesApi = new SitesApi(alfrescoApi);
        this.activitiesApi = new ActivitiesApi(alfrescoApi);
        this.preferencesApi = new PreferencesApi(alfrescoApi);
        this.groupsApi = new GroupsApi(alfrescoApi);
    };
    PeopleApi.prototype.addFavorite = function (personId, favoriteBody) {
        return this.favoritesApi.createFavorite(personId, favoriteBody);
    };
    PeopleApi.prototype.addSiteMembershipRequest = function (personId, siteMembershipBody) {
        return this.sitesApi.createSiteMembershipRequestForPerson(personId, siteMembershipBody);
    };
    PeopleApi.prototype.deleteFavoriteSite = function (personId, siteId) {
        return this.favoritesApi.deleteSiteFavorite(personId, siteId);
    };
    PeopleApi.prototype.favoriteSite = function (personId, favoriteSiteBody) {
        return this.favoritesApi.createSiteFavorite(personId, favoriteSiteBody);
    };
    PeopleApi.prototype.getActivities = function (personId, opts) {
        return this.activitiesApi.listActivitiesForPerson(personId, opts);
    };
    PeopleApi.prototype.getFavorite = function (personId, favoriteId, opts) {
        return this.favoritesApi.getFavorite(personId, favoriteId, opts);
    };
    PeopleApi.prototype.getFavoriteSite = function (personId, siteId, opts) {
        return this.favoritesApi.getFavoriteSite(personId, siteId, opts);
    };
    PeopleApi.prototype.getFavoriteSites = function (personId, opts) {
        return this.favoritesApi.listFavoriteSitesForPerson(personId, opts);
    };
    PeopleApi.prototype.getFavorites = function (personId, opts) {
        return this.favoritesApi.listFavorites(personId, opts);
    };
    PeopleApi.prototype.getPerson = function (personId, opts) {
        return this.peopleApiNew.getPerson(personId, opts);
    };
    PeopleApi.prototype.getPersons = function (opts) {
        return this.peopleApiNew.listPeople(opts);
    };
    PeopleApi.prototype.addPerson = function (person) {
        return this.peopleApiNew.createPerson(person);
    };
    PeopleApi.prototype.updatePerson = function (personId, personBodyUpdate, opts) {
        return this.peopleApiNew.updatePerson(personId, personBodyUpdate, opts);
    };
    PeopleApi.prototype.getPersonNetwork = function (personId, networkId, opts) {
        return this.networksApi.getNetworkForPerson(personId, networkId, opts);
    };
    PeopleApi.prototype.getPersonNetworks = function (personId, opts) {
        return this.networksApi.listNetworksForPerson(personId, opts);
    };
    PeopleApi.prototype.getPreference = function (personId, preferenceName, opts) {
        return this.preferencesApi.getPreference(personId, preferenceName, opts);
    };
    PeopleApi.prototype.getPreferences = function (personId, opts) {
        return this.preferencesApi.listPreferences(personId, opts);
    };
    PeopleApi.prototype.getSiteMembership = function (personId, opts) {
        return this.sitesApi.listSiteMembershipsForPerson(personId, opts);
    };
    PeopleApi.prototype.getGroupsMembership = function (personId, opts) {
        return this.groupsApi.listGroupMembershipsForPerson(personId, opts);
    };
    PeopleApi.prototype.getSiteMembershipRequest = function (personId, siteId, opts) {
        return this.sitesApi.getSiteMembershipRequestForPerson(personId, siteId, opts);
    };
    PeopleApi.prototype.getSiteMembershipRequests = function (personId, opts) {
        return this.sitesApi.listSiteMembershipRequestsForPerson(personId, opts);
    };
    PeopleApi.prototype.removeFavoriteSite = function (personId, favoriteId) {
        return this.favoritesApi.deleteSiteFavorite(personId, favoriteId);
    };
    PeopleApi.prototype.listSiteMembershipsForPerson = function (personId, opts) {
        return this.sitesApi.listSiteMembershipsForPerson(personId, opts);
    };
    PeopleApi.prototype.removeSiteMembershipRequest = function (personId, siteId) {
        return this.sitesApi.deleteSiteMembershipRequestForPerson(personId, siteId);
    };
    PeopleApi.prototype.updateSiteMembershipRequest = function (personId, siteId, siteMembershipBody) {
        return this.sitesApi.updateSiteMembershipRequestForPerson(personId, siteId, siteMembershipBody);
    };
    return PeopleApi;
}());
export { PeopleApi };
