// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/api/ratingsApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { RatingsApi as NewRatingsApi } from '../../../../api/content-rest-api/api/ratings.api';
var RatingsApi = (function () {
    function RatingsApi() {
    }
    RatingsApi.prototype.init = function (alfrescoApi) {
        this.ratingsApi = new NewRatingsApi(alfrescoApi);
    };
    RatingsApi.prototype.getRating = function (nodeId, ratingId, opts) {
        return this.ratingsApi.getRating(nodeId, ratingId, opts);
    };
    RatingsApi.prototype.getRatings = function (nodeId, opts) {
        return this.ratingsApi.listRatings(nodeId, opts);
    };
    RatingsApi.prototype.rate = function (nodeId, ratingBody) {
        return this.ratingsApi.createRating(nodeId, ratingBody);
    };
    RatingsApi.prototype.removeRating = function (nodeId, ratingId) {
        return this.ratingsApi.deleteRating(nodeId, ratingId);
    };
    return RatingsApi;
}());
export { RatingsApi };
