// @alfresco/js-api/esm5/src/api-legacy/content-rest-api/src/api/tagsApi.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { TagsApi as NewTagsApi } from '../../../../api/content-rest-api/api/tags.api';
var TagsApi = (function () {
    function TagsApi() {
    }
    TagsApi.prototype.init = function (alfrescoApi) {
        this.tagsApi = new NewTagsApi(alfrescoApi);
    };
    TagsApi.prototype.addTag = function (nodeId, tagBody) {
        return this.tagsApi.createTagForNode(nodeId, tagBody);
    };
    TagsApi.prototype.getNodeTags = function (nodeId, opts) {
        return this.tagsApi.listTagsForNode(nodeId, opts);
    };
    TagsApi.prototype.getTag = function (tagId, opts) {
        return this.tagsApi.getTag(tagId, opts);
    };
    TagsApi.prototype.getTags = function (opts) {
        return this.tagsApi.listTags(opts);
    };
    TagsApi.prototype.removeTag = function (nodeId, tagId) {
        return this.tagsApi.deleteTagFromNode(nodeId, tagId);
    };
    TagsApi.prototype.updateTag = function (tagId, tagBody) {
        return this.tagsApi.updateTag(tagId, tagBody);
    };
    return TagsApi;
}());
export { TagsApi };
