// @alfresco/js-api/esm2015/src/api/content-rest-api/api/base.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
export class BaseApi {
    constructor(alfrescoApi) {
        this.alfrescoApi = alfrescoApi;
    }
    get apiClient() {
        return this.alfrescoApi.contentClient;
    }
    post(options) {
        return this.apiClient.post(options);
    }
    put(options) {
        return this.apiClient.put(options);
    }
    get(options) {
        return this.apiClient.get(options);
    }
    delete(options) {
        return this.apiClient.delete(options);
    }
    errorMessage(param, methodName) {
        return `Missing param ${param} in ${methodName}`;
    }
}
