// @alfresco/js-api/src/contentClient.js
"use strict";
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContentClient = void 0;
const alfrescoApiClient_1 = require("./alfrescoApiClient");
class ContentClient extends alfrescoApiClient_1.AlfrescoApiClient {
    constructor(config, servicePath) {
        super();
        this.className = 'ContentClient';
        this.setConfig(config, servicePath);
    }
    setConfig(config, servicePath) {
        this.config = config;
        this.servicePath = servicePath;
        this.changeHost();
    }
    changeHost() {
        this.host = this.config.hostEcm;
        this.basePath = `${this.config.hostEcm}/${this.config.contextRoot}${this.servicePath}`;
    }
    setAuthentications(authentications) {
        this.authentications = authentications;
    }
}
exports.ContentClient = ContentClient;
