// @alfresco/js-api/esm5/src/api/gs-core-rest-api/api/unfiledRecordFolders.api.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { __extends } from "tslib";
import { UnfiledRecordFolderAssociationPaging } from '../model/unfiledRecordFolderAssociationPaging';
import { UnfiledRecordFolderEntry } from '../model/unfiledRecordFolderEntry';
import { BaseApi } from './base.api';
import { buildCollectionParam } from '../../../alfrescoApiClient';
import { throwIfNotDefined } from '../../../assert';
var UnfiledRecordFoldersApi = (function (_super) {
    __extends(UnfiledRecordFoldersApi, _super);
    function UnfiledRecordFoldersApi() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    UnfiledRecordFoldersApi.prototype.createUnfiledRecordFolderChildren = function (unfiledRecordFolderId, nodeBodyCreate, opts) {
        throwIfNotDefined(unfiledRecordFolderId, 'unfiledRecordFolderId');
        throwIfNotDefined(nodeBodyCreate, 'nodeBodyCreate');
        opts = opts || {};
        var postBody = nodeBodyCreate;
        var pathParams = {
            'unfiledRecordFolderId': unfiledRecordFolderId
        };
        var queryParams = {
            'autoRename': opts['autoRename'],
            'include': buildCollectionParam(opts['include'], 'csv'),
            'fields': buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json', 'multipart/form-data'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/unfiled-record-folders/{unfiledRecordFolderId}/children', 'POST', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, UnfiledRecordFolderAssociationPaging);
    };
    UnfiledRecordFoldersApi.prototype.deleteUnfiledRecordFolder = function (unfiledRecordFolderId) {
        throwIfNotDefined(unfiledRecordFolderId, 'unfiledRecordFolderId');
        var postBody = null;
        var pathParams = {
            'unfiledRecordFolderId': unfiledRecordFolderId
        };
        var queryParams = {};
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/unfiled-record-folders/{unfiledRecordFolderId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts);
    };
    UnfiledRecordFoldersApi.prototype.getUnfiledRecordFolder = function (unfiledRecordFolderId, opts) {
        throwIfNotDefined(unfiledRecordFolderId, 'unfiledRecordFolderId');
        opts = opts || {};
        var postBody = null;
        var pathParams = {
            'unfiledRecordFolderId': unfiledRecordFolderId
        };
        var queryParams = {
            'include': buildCollectionParam(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'fields': buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/unfiled-record-folders/{unfiledRecordFolderId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, UnfiledRecordFolderEntry);
    };
    UnfiledRecordFoldersApi.prototype.listUnfiledRecordFolderChildren = function (unfiledRecordFolderId, opts) {
        throwIfNotDefined(unfiledRecordFolderId, 'unfiledRecordFolderId');
        opts = opts || {};
        var postBody = null;
        var pathParams = {
            'unfiledRecordFolderId': unfiledRecordFolderId
        };
        var queryParams = {
            'skipCount': opts['skipCount'],
            'maxItems': opts['maxItems'],
            'where': opts['where'],
            'include': buildCollectionParam(opts['include'], 'csv'),
            'relativePath': opts['relativePath'],
            'includeSource': opts['includeSource'],
            'fields': buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/unfiled-record-folders/{unfiledRecordFolderId}/children', 'GET', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, UnfiledRecordFolderAssociationPaging);
    };
    UnfiledRecordFoldersApi.prototype.updateUnfiledRecordFolder = function (unfiledRecordFolderId, unfiledRecordFolderBodyUpdate, opts) {
        throwIfNotDefined(unfiledRecordFolderId, 'unfiledRecordFolderId');
        throwIfNotDefined(unfiledRecordFolderBodyUpdate, 'unfiledRecordFolderBodyUpdate');
        opts = opts || {};
        var postBody = unfiledRecordFolderBodyUpdate;
        var pathParams = {
            'unfiledRecordFolderId': unfiledRecordFolderId
        };
        var queryParams = {
            'include': buildCollectionParam(opts['include'], 'csv'),
            'includeSource': opts['includeSource'],
            'fields': buildCollectionParam(opts['fields'], 'csv')
        };
        var headerParams = {};
        var formParams = {};
        var contentTypes = ['application/json'];
        var accepts = ['application/json'];
        return this.apiClient.callApi('/unfiled-record-folders/{unfiledRecordFolderId}', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, contentTypes, accepts, UnfiledRecordFolderEntry);
    };
    return UnfiledRecordFoldersApi;
}(BaseApi));
export { UnfiledRecordFoldersApi };
