// @alfresco/js-api/esm5/src/api/search-rest-api/model/resultSetContext.js
/*!
* @license
* Copyright 2018 Alfresco Software, Ltd.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
import { GenericFacetResponse } from './genericFacetResponse';
import { ResponseConsistency } from './responseConsistency';
import { ResultBuckets } from './resultBuckets';
import { ResultSetContextFacetQueries } from './resultSetContextFacetQueries';
import { ResultSetContextSpellcheck } from './resultSetContextSpellcheck';
import { SearchRequest } from './searchRequest';
var ResultSetContext = (function () {
    function ResultSetContext(input) {
        if (input) {
            Object.assign(this, input);
            this.consistency = input.consistency ? new ResponseConsistency(input.consistency) : undefined;
            this.request = input.request ? new SearchRequest(input.request) : undefined;
            if (input.facetQueries) {
                this.facetQueries = input.facetQueries.map(function (item) {
                    return new ResultSetContextFacetQueries(item);
                });
            }
            if (input.facetsFields) {
                this.facetsFields = input.facetsFields.map(function (item) {
                    return new ResultBuckets(item);
                });
            }
            if (input.facets) {
                this.facets = input.facets.map(function (item) {
                    return new GenericFacetResponse(item);
                });
            }
            if (input.spellcheck) {
                this.spellcheck = input.spellcheck.map(function (item) {
                    return new ResultSetContextSpellcheck(item);
                });
            }
        }
    }
    return ResultSetContext;
}());
export { ResultSetContext };
